tools/kafkatool/dump.go

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.uber.org/atomic"
+)
+
+type DumpCommand struct {
+	topic             string
+	partition         int
+	skipFirst         int
+	exportOffsetStart int64
+	exportMaxRecords  int
+	inOutFile         *os.File
+	getKafkaClient    func() *kgo.Client
+	printer           Printer
+}
+
+// Register is used to register the command to a parent command.
+func (c *DumpCommand) Register(app *kingpin.Application, getKafkaClient func() *kgo.Client, printer Printer) {
+	c.getKafkaClient = getKafkaClient
+	c.printer = printer
+
+	cmd := app.Command("dump", "Dump Kafka topic contents")
+	cmd.Flag("topic", "Kafka topic to dump").Required().StringVar(&c.topic)
+	cmd.Flag("partition", "Kafka partition to dump or import into").Required().IntVar(&c.partition)
+	cmd.Flag("skip-first", "Skip until input record with offset N").Default("0").IntVar(&c.skipFirst)
+	cmd.Flag("offset", "Offset to start exporting from").Default("0").Int64Var(&c.exportOffsetStart)
+	cmd.Flag("export-max-records", "Maximum number of records to export").Default("1000000").IntVar(&c.exportMaxRecords)
+	cmd.Flag("file", "File to read from or write to.").Required().OpenFileVar(&c.inOutFile, os.O_RDWR|os.O_CREATE, 0600)
+
+	cmd.Command("import", "Import records from a file into a Kafka topic").Action(c.doImport)
+	cmd.Command("export", "Export records from a Kafka topic into a file").Action(c.doExport)
+}
+
+type key int
+
+const (
+	originalOffsetKey key = iota
+)
+
+func (c *DumpCommand) doExport(*kingpin.ParseContext) error {
+	client := c.getKafkaClient()
+	client.AddConsumePartitions(map[string]map[int32]kgo.Offset{
+		c.topic: {int32(c.partition): kgo.NewOffset().At(c.exportOffsetStart)}},
+	)
+	go func() {
+		for {
+			time.Sleep(time.Second)
+			c.printer.PrintLine(fmt.Sprintf("produced records: %d, offset %d", recordCount.Load(), consumedOffset.Load()))
+		}
+	}()
+
+	encoder := json.NewEncoder(c.inOutFile)
+
+	lastAvailableOffset := int64(-1)
+	for recordCount.Load() < int64(c.exportMaxRecords) {
+		fetches := client.PollFetches(context.Background())
+		if err := fetches.Err(); err != nil {
+			return fmt.Errorf("failed to fetch records: %w", err)
+		}
+		fetches.EachPartition(func(partition kgo.FetchTopicPartition) {
+			lastAvailableOffset = max(lastAvailableOffset, partition.HighWatermark-1)
+		})
+		var err error
+		fetches.EachRecord(func(record *kgo.Record) {
+			if recordCount.Inc() > int64(c.exportMaxRecords) {
+				return
+			}
+			consumedOffset.Store(record.Offset)
+			encodeErr := encoder.Encode(record)
```

Comment:
Could you explain, how does json encoding a record with protobuf data work — does it produce a long hex-encoded string?

Reply:
it's just json encoding of a struct, nothing fancy. the json encoder would encode `[]byte` as base64-encoded strings in the output. You get output then looks like this 


<details><summary>Details</summary>
<p>

```json
{
  "Key": "MTc5MjM=",
  "Value": "CqIECi4KCF9fbmFtZV9fEiJrZXBsZXJfY29udGFpbmVyX2RyYW1fam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA3YzEwYTBlZmEzNDE0NDcwNjVmMmQ5NTIwYWVjODU5MGUyMzAwYzkyMjk3NWRiZDA3NDljNDE3YzBmZTg2OGJiChwKDmNvbnRhaW5lcl9uYW1lEgprdWJlLXByb3h5CiIKE2NvbnRhaW5lcl9uYW1lc3BhY2USC2t1YmUtc3lzdGVtCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDAoEbW9kZRIEaWRsZQoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgpKCghwb2RfbmFtZRI+a3ViZS1wcm94eS1na2UtZGV2LXVzLWNlbnRyYWwtMC1tYWluLW4yczE2LTMtMWRkLTliNTAyZDk2LXR2bWsKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJsHJokbF0+EAQ+IP4nJAyCv0DCjIKCF9fbmFtZV9fEiZrZXBsZXJfY29udGFpbmVyX3BsYXRmb3JtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAMTU0ODEyMzk5MmQ3MGEzMTFhNWVlOWFjMGNkMGNhN2NmNmMzMGEyNmE0Mjc1MjUxNmRjOWQwN2ZiMzdlNjQ4NQoaCg5jb250YWluZXJfbmFtZRIIaW5nZXN0ZXIKIwoTY29udGFpbmVyX25hbWVzcGFjZRIMbWltaXItZGV2LTExCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgofCghwb2RfbmFtZRITaW5nZXN0ZXItem9uZS1jLTEwMQodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAnb+X5q5A3VQBD4g/ickDIKzAIKNgoIX19uYW1lX18SKmNvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9tZXRhZGF0YV90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxNDEzNRIHEPKC+JyQMgqDAwpFCghfX25hbWVfXxI5bWVtYmVybGlzdF9jbGllbnRfa3Zfc3RvcmVfdmFsdWVfdG9tYnN0b25lc19yZW1vdmVkX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKLAoDa2V5EiVydWxlci1xdWVyeS1zY2hlZHVsZXIvcXVlcnktc2NoZWR1bGVyChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAAAAAPA/EPKC+JyQMgrLAgosCghfX25hbWVfXxIgcHJvY2Vzc192aXJ0dWFsX21lbW9yeV9tYXhfYnl0ZXMKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAobCgljb250YWluZXISDmxvZ3MtZm9yd2FyZGVyCkcKCGluc3RhbmNlEjtsb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmOmxvZ3MtZm9yd2FyZGVyOmh0dHAtbWV0cmljcwoiCgNqb2ISG2luc2lnaHQtbG9ncy9sb2dzLWZvcndhcmRlcgoZCgluYW1lc3BhY2USDGluc2lnaHQtbG9ncwomCgNwb2QSH2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2YSEAkAAAAAAADwQxDagvickDIKuQMKNQoIX19uYW1lX18SKWNvcnRleF9rdl9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKIAoHa3ZfbmFtZRIVZGlzdHJpYnV0b3ItaGF0cmFja2VyCgsKAmxlEgUwLjAwNQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA0dFVAojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDwoEcm9sZRIHcHJpbWFyeQoSCgtzdGF0dXNfY29kZRIDMjAwCgwKBHR5cGUSBGV0Y2QSEAkAAAAAALSSQBDygvickDIKxgIKKwoIX19uYW1lX18SH2NvcnRleF9sYWJlbHNfcGVyX3NhbXBsZV9idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoJCgJsZRIDNS4wChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAFBSt6hBEPKC+JyQMgrZAgo6CghfX25hbWVfXxIuY29ydGV4X3NlbGZfbW9uaXRvcmluZ19kaXNrX3JlYWRzX21lcmdlZF90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKDQoGZGV2aWNlEgNzZGcKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAAAA+qxAEPKC+JyQMgrVAgo2CghfX25hbWVfXxIqY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX3JlcXVlc3RzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTEwOTcxEhAJAAAAAAAACEAQ8oL4nJAyCs0CCjcKCF9fbmFtZV9fEitjb3J0ZXhfZGlzdHJpYnV0b3JfcmVjZWl2ZWRfZXhlbXBsYXJzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTEwNzk2EgcQ8oL4nJAyCvoCCjAKCF9fbmFtZV9fEiRjb3J0ZXhfcmVzcG9uc2VfbWVzc2FnZV9ieXRlc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoRCgJsZRILMi42MjE0NGUrMDYKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwChYKBXJvdXRlEg1pbmdlc3Rlcl9yaW5nEhAJAAAAAAAA8D8Q8oL4nJAyCrgCChMKCF9fbmFtZV9fEgdnb19pbmZvCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwChMKB3ZlcnNpb24SCGdvMS4yMi40EhAJAAAAAAAA8D8Q8oL4nJAyCuIDCi0KCF9fbmFtZV9fEiFrZXBsZXJfY29udGFpbmVyX2dwdV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDE0NTY3NzUxYjU2ZDUzOTU3ZjZiMjdiNmEwYTU2Zjg4MjkxZjA5ZWMwNjE3MWE3NzhlZGRhZGQyNjE4OTIwYWIKHwoOY29udGFpbmVyX25hbWUSDWlwLW1hc3EtYWdlbnQKIgoTY29udGFpbmVyX25hbWVzcGFjZRILa3ViZS1zeXN0ZW0KNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh8KCHBvZF9uYW1lEhNpcC1tYXNxLWFnZW50LTc5N2NuCg8KBnNvdXJjZRIFZHVtbXkSBxD4g/ickDIK0gIKPAoIX19uYW1lX18SMGphZWdlcl90cmFjZXJfc3Bhbl9jb250ZXh0X2RlY29kaW5nX2Vycm9yc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZhIHENqC+JyQMgq2Awo6CghfX25hbWVfXxIudGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCggKAmxlEgIxMAofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoYCglvcGVyYXRpb24SC2NvbXBhcmVzd2FwCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIHEPKC+JyQMgqfAwpCCghfX25hbWVfXxI2Y29ydGV4X2luZ2VzdGVyX2NsaWVudF9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCwoCbGUSBTAuMjU2ChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiIKCW9wZXJhdGlvbhIVL2NvcnRleC5Jbmdlc3Rlci9QdXNoCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoaCgtzdGF0dXNfY29kZRILVW5hdmFpbGFibGUSEAkAAAAAAI3bQBDygvickDIK4QIKMgoIX19uYW1lX18SJmthZmthX3dyaXRlcl93cml0ZV90aW1lX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCwoCbGUSBTAuMDA4ChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChAKB25vZGVfaWQSBTIwMDAwCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAADcNRVBEPKC+JyQMgqzAwo6CghfX25hbWVfXxIudGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUwLjAyNQofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoSCglvcGVyYXRpb24SBXRvdWNoCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIHEPKC+JyQMgruAgooCghfX25hbWVfXxIccmVxdWVzdF9tZXNzYWdlX2J5dGVzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChUKCWNvbnRhaW5lchIIcHJvbXRhaWwKPQoIaW5zdGFuY2USMXByb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ejpwcm9tdGFpbDpodHRwLW1ldHJpY3MKKQoDam9iEiJsb2tpLWRldi1zc2QvcHJvbXRhaWwtbG9raS1kZXYtc3NkChEKAmxlEgs1LjI0Mjg4ZSswNgoNCgZtZXRob2QSA0dFVAoZCgluYW1lc3BhY2USDGxva2ktZGV2LXNzZAoiCgNwb2QSG3Byb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6egoQCgVyb3V0ZRIHbWV0cmljcxIQCQAAAAAAmqRAEOKD+JyQMgr4AgowCghfX25hbWVfXxIkY29ydGV4X3Jlc3BvbnNlX21lc3NhZ2VfYnl0ZXNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKEQoCbGUSCzUuMjQyODhlKzA3Cg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoUCgVyb3V0ZRILZGVidWdfcHByb2YSEAkAAAAA0DvzQBDygvickDIK3AIKPQoIX19uYW1lX18SMWNvcnRleF9zZWxmX21vbml0b3JpbmdfZGlza19pb190aW1lX3NlY29uZHNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCg0KBmRldmljZRIDc2RjCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkbL90kBjCQQBDygvickDIKnwIKJQoIX19uYW1lX18SGWxva2lfY2FuYXJ5X2VudHJpZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2xva2ktY2FuYXJ5CjYKCGluc3RhbmNlEipsb2tpLWNhbmFyeS1oeGZrZDpsb2tpLWNhbmFyeTpodHRwLW1ldHJpY3MKHwoDam9iEhhwcm9tdGFpbC1vcHMvbG9raS1jYW5hcnkKGQoJbmFtZXNwYWNlEgxwcm9tdGFpbC1vcHMKGAoDcG9kEhFsb2tpLWNhbmFyeS1oeGZrZBIQCQAAAADQ1PdAEKiD+JyQMgrWAwoyCghfX25hbWVfXxIma2VwbGVyX2NvbnRhaW5lcl9icGZfY3B1X3RpbWVfbXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQGVmYjAwMGNjNjc3ZWIzOTE3MTZhZmEwNzRiYmM2NDIyY2I1OGY3YmYwNTMzZjA2NDVlOWJmZGUzYjM0N2Y0NTIKFwoOY29udGFpbmVyX25hbWUSBXdhdGNoCiUKE2NvbnRhaW5lcl9uYW1lc3BhY2USDnB5cm9zY29wZS1lYnBmCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKGgoIcG9kX25hbWUSDnByb2ZpbGVyLTluYnI5Cg0KBnNvdXJjZRIDYnBmEhAJAAAAAAAAJkAQ+IP4nJAyCqoDCjIKCF9fbmFtZV9fEiZjb3J0ZXhfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCggKAmxlEgIxMAoOCgZtZXRob2QSBGdSUEMKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCiYKBXJvdXRlEh0vZGlzdHJpYnV0b3IuRGlzdHJpYnV0b3IvUHVzaAoXCgtzdGF0dXNfY29kZRIISW50ZXJuYWwKCwoCd3MSBWZhbHNlEhAJAAAAAAAA8D8Q8oL4nJAyCoYDCjIKCF9fbmFtZV9fEiZjb3J0ZXhfZGlzdHJpYnV0b3Jfc2FtcGxlc19wZXJfcmVxdWVzdAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoMCgR1c2VyEgQ5OTYwIkYIzMGRAxkAADBm4r4HQimdl4zPCLpbJTD+a1oCEA5iH8wHL48BlgWcBGs+lgroFexWjNcg8O/ZBfX5+QXDwAF48oL4nJAyCoQECjEKCF9fbmFtZV9fEiVrZXBsZXJfY29udGFpbmVyX3BhY2thZ2Vfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA5ZjI2ZmMzNTRhMThlY2U0Njk3MGY1ZjMxZjFmNGJhOWVjNGZhMTg4OTRiNGZmMmYwYWQzMzExZThiMTA4YzI3ChoKDmNvbnRhaW5lcl9uYW1lEghwcm9tdGFpbAojChNjb250YWluZXJfbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCicKCHBvZF9uYW1lEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtZDZkZGMKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJarx0kxiEYkAQ+IP4nJAyCrIDCjoKCF9fbmFtZV9fEi5sb2dzX2ZvcndhcmRlcl9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGwoJY29udGFpbmVyEg5sb2dzLWZvcndhcmRlcgpHCghpbnN0YW5jZRI7bG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZjpsb2dzLWZvcndhcmRlcjpodHRwLW1ldHJpY3MKIgoDam9iEhtpbnNpZ2h0LWxvZ3MvbG9ncy1mb3J3YXJkZXIKCwoCbGUSBTAuMDI1Cg4KBm1ldGhvZBIEUE9TVAoZCgluYW1lc3BhY2USDGluc2lnaHQtbG9ncwomCgNwb2QSH2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2YKGQoFcm91dGUSEGxva2lfYXBpX3YxX3B1c2gKEgoLc3RhdHVzX2NvZGUSAzUwMAoLCgJ3cxIFZmFsc2USEAkAAAAAAADwPxDagvickDIKgQMKOgoIX19uYW1lX18SLmdyYWZhbmFjb21fYXBpX3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoLCgRjb2RlEgMyMDAKGwoJY29udGFpbmVyEg5sb2dzLWZvcndhcmRlcgpHCghpbnN0YW5jZRI7bG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZjpsb2dzLWZvcndhcmRlcjpodHRwLW1ldHJpY3MKIgoDam9iEhtpbnNpZ2h0LWxvZ3MvbG9ncy1mb3J3YXJkZXIKCwoCbGUSBTAuMDI1ChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzChUKBHBhdGgSDWhvc3RlZC1hbGVydHMKJgoDcG9kEh9sb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmEgcQ2oL4nJAyCt8CCkAKCF9fbmFtZV9fEjRjb3J0ZXhfc2VsZl9tb25pdG9yaW5nX2Rpc2tfd3JpdGVfdGltZV9zZWNvbmRzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgoNCgZkZXZpY2USA3NkYwpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJ0SLb+X77lUAQ8oL4nJAyCokDCjgKCF9fbmFtZV9fEixsb2dzX2ZvcndhcmRlcl9yZXNwb25zZV9tZXNzYWdlX2J5dGVzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChIKAmxlEgwxLjA0ODU3NmUrMDcKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxpbnNpZ2h0LWxvZ3MKJgoDcG9kEh9sb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmCg0KBXJvdXRlEgRyb290EhAJAAAAACgeBUEQ2oL4nJAyCvwDCjEKCF9fbmFtZV9fEiVrZXBsZXJfY29udGFpbmVyX3BhY2thZ2Vfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkAxNTQ4MTIzOTkyZDcwYTMxMWE1ZWU5YWMwY2QwY2E3Y2Y2YzMwYTI2YTQyNzUyNTE2ZGM5ZDA3ZmIzN2U2NDg1ChoKDmNvbnRhaW5lcl9uYW1lEghpbmdlc3RlcgojChNjb250YWluZXJfbmFtZXNwYWNlEgxtaW1pci1kZXYtMTEKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh8KCHBvZF9uYW1lEhNpbmdlc3Rlci16b25lLWMtMTAxCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCf7UeOk2FNBAEPiD+JyQMgrcAgo9CghfX25hbWVfXxIxY29ydGV4X3NlbGZfbW9uaXRvcmluZ19kaXNrX3JlYWRzX2NvbXBsZXRlZF90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKDQoGZGV2aWNlEgNzZGIKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAGCIc2dBEPKC+JyQMgrMAgo2CghfX25hbWVfXxIqY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX21ldGFkYXRhX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTE0Mzk5EgcQ8oL4nJAyCosECjAKCF9fbmFtZV9fEiRrZXBsZXJfY29udGFpbmVyX3VuY29yZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQGQ5NTdmZGE0OGNhMzQ3OGVmYjUwYzg2MTZhZTMxZDJhNGY0ZWQxZGI0Yzc1MDM4MGI2NzMxZDczMjE1ODBlMGIKJgoOY29udGFpbmVyX25hbWUSFGdyYWZhbmEtY29tLWV4cG9ydGVyCiIKE2NvbnRhaW5lcl9uYW1lc3BhY2USC2dyYWZhbmEtY29tCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDAoEbW9kZRIEaWRsZQoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgowCghwb2RfbmFtZRIkZ3JhZmFuYS1jb20tZXhwb3J0ZXItMS05YjY2NWQ5LWxsajJ3Ch0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIHEPiD+JyQMgqdAgojCghfX25hbWVfXxIXZ29fbWVtc3RhdHNfYWxsb2NfYnl0ZXMKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2xva2ktY2FuYXJ5CjYKCGluc3RhbmNlEipsb2tpLWNhbmFyeS1oeGZrZDpsb2tpLWNhbmFyeTpodHRwLW1ldHJpY3MKHwoDam9iEhhwcm9tdGFpbC1vcHMvbG9raS1jYW5hcnkKGQoJbmFtZXNwYWNlEgxwcm9tdGFpbC1vcHMKGAoDcG9kEhFsb2tpLWNhbmFyeS1oeGZrZBIQCQAAAACEGlxBEKiD+JyQMgqwAwo6CghfX25hbWVfXxIubG9nc19mb3J3YXJkZXJfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyCgkKAmxlEgMwLjUKDgoGbWV0aG9kEgRQT1NUChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoZCgVyb3V0ZRIQbG9raV9hcGlfdjFfcHVzaAoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAOj2oYlBENqC+JyQMgrcAgotCghfX25hbWVfXxIhY29ydGV4X3Jlc3BvbnNlX21lc3NhZ2VfYnl0ZXNfc3VtCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg4KBXJvdXRlEgVyZWFkeRIQCQAAAADoChJBEPKC+JyQMgrHAwopCghfX25hbWVfXxIda2VwbGVyX2NvbnRhaW5lcl9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQGJkYjU1ZDkyYjZmZjM4NTBkYTljYmFhODJlNGJlMTU2MDY4YzMzNzNlMWY2OGUzYWU5NGRkMGU4YjRhOGIyNjkKFwoOY29udGFpbmVyX25hbWUSBWFnZW50CiMKE2NvbnRhaW5lcl9uYW1lc3BhY2USDHByb210YWlsLW9wcwo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKFwoIcG9kX25hbWUSC2FnZW50LWdwczJmEhAJkxgEVg3gC0EQ+IP4nJAyCuACCjIKCF9fbmFtZV9fEiZrYWZrYV93cml0ZXJfd3JpdGVfdGltZV9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgoKAmxlEgQrSW5mChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChAKB25vZGVfaWQSBTIwMDAwCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAAD8NRVBEPKC+JyQMgqJAwo6CghfX25hbWVfXxIuZ3JhZmFuYWNvbV9hcGlfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wCgsKBGNvZGUSAzIwMAobCgljb250YWluZXISDmxvZ3MtZm9yd2FyZGVyCkcKCGluc3RhbmNlEjtsb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmOmxvZ3MtZm9yd2FyZGVyOmh0dHAtbWV0cmljcwoiCgNqb2ISG2luc2lnaHQtbG9ncy9sb2dzLWZvcndhcmRlcgoJCgJsZRIDMC41ChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzChYKBHBhdGgSDmhvc3RlZC1tZXRyaWNzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZhIQCQAAAAAACrhAENqC+JyQMgq2AgovCghfX25hbWVfXxIjbWVtYmVybGlzdF9jbGllbnRfY2FzX2ZhaWx1cmVfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASBxDygvickDIKoAIKJgoIX19uYW1lX18SGnByb2Nlc3Nfc3RhcnRfdGltZV9zZWNvbmRzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtsb2tpLWNhbmFyeQo2CghpbnN0YW5jZRIqbG9raS1jYW5hcnktaHhma2Q6bG9raS1jYW5hcnk6aHR0cC1tZXRyaWNzCh8KA2pvYhIYcHJvbXRhaWwtb3BzL2xva2ktY2FuYXJ5ChkKCW5hbWVzcGFjZRIMcHJvbXRhaWwtb3BzChgKA3BvZBIRbG9raS1jYW5hcnktaHhma2QSEAmF6zGBB6rZQRCog/ickDIKkwMKPwoIX19uYW1lX18SM2NvcnRleF9pbmdlc3Rlcl9jbGllbnRfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX3N1bQofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiIKCW9wZXJhdGlvbhIVL2NvcnRleC5Jbmdlc3Rlci9QdXNoCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoeCgtzdGF0dXNfY29kZRIPSW52YWxpZEFyZ3VtZW50EhAJWVyoPqAje0AQ8oL4nJAyCokDCjoKCF9fbmFtZV9fEi5ncmFmYW5hY29tX2FwaV9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKCwoEY29kZRIDMjAwChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyCgoKAmxlEgQrSW5mChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzChUKBHBhdGgSDWhvc3RlZC10cmFjZXMKJgoDcG9kEh9sb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmEhAJAAAAAABIt0AQ2oL4nJAyCrwCCjMKCF9fbmFtZV9fEidsb2tpX2V4cGVyaW1lbnRhbF9mZWF0dXJlc19pbl91c2VfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoVCgljb250YWluZXISCHByb210YWlsCj0KCGluc3RhbmNlEjFwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42eno6cHJvbXRhaWw6aHR0cC1tZXRyaWNzCikKA2pvYhIibG9raS1kZXYtc3NkL3Byb210YWlsLWxva2ktZGV2LXNzZAoZCgluYW1lc3BhY2USDGxva2ktZGV2LXNzZAoiCgNwb2QSG3Byb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ehIHEOKD+JyQMgrpAwotCghfX25hbWVfXxIha2VwbGVyX2NvbnRhaW5lcl9ncHVfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA5ZjI2ZmMzNTRhMThlY2U0Njk3MGY1ZjMxZjFmNGJhOWVjNGZhMTg4OTRiNGZmMmYwYWQzMzExZThiMTA4YzI3ChoKDmNvbnRhaW5lcl9uYW1lEghwcm9tdGFpbAojChNjb250YWluZXJfbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCicKCHBvZF9uYW1lEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtZDZkZGMKDwoGc291cmNlEgVkdW1teRIHEPiD+JyQMgrTAgoiCghfX25hbWVfXxIWZ29fZ2NfZHVyYXRpb25fc2Vjb25kcwofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoQCghxdWFudGlsZRIEMC43NRIQCRLeB+gKKx8/ENqC+JyQMgrXAgo0CghfX25hbWVfXxIoY29ydGV4X2Rpc3RyaWJ1dG9yX2V4ZW1wbGFyc19wZXJfcmVxdWVzdAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxMDk3MSIUCAMpnZeMzwi6WyUwA3jygvickDIKzAIKNgoIX19uYW1lX18SKmNvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9tZXRhZGF0YV90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxOTA1NRIHEPKC+JyQMgrhAgoyCghfX25hbWVfXxIma2Fma2Ffd3JpdGVyX3dyaXRlX3RpbWVfc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoLCgJsZRIFMC4wNjQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKEAoHbm9kZV9pZBIFMjAwMDAKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAPw1FUEQ8oL4nJAyCpsDCkIKCF9fbmFtZV9fEjZjb3J0ZXhfaW5nZXN0ZXJfY2xpZW50X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoLCgJsZRIFMC4wMTYKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIgoJb3BlcmF0aW9uEhUvY29ydGV4LkluZ2VzdGVyL1B1c2gKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCh8KC3N0YXR1c19jb2RlEhBEZWFkbGluZUV4Y2VlZGVkEgcQ8oL4nJAyCuYDCi0KCF9fbmFtZV9fEiFrZXBsZXJfY29udGFpbmVyX2dwdV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDlmMjZmYzM1NGExOGVjZTQ2OTcwZjVmMzFmMWY0YmE5ZWM0ZmExODg5NGI0ZmYyZjBhZDMzMTFlOGIxMDhjMjcKGgoOY29udGFpbmVyX25hbWUSCHByb210YWlsCiMKE2NvbnRhaW5lcl9uYW1lc3BhY2USDGxva2ktZGV2LXNzZAo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKJwoIcG9kX25hbWUSG3Byb210YWlsLWxva2ktZGV2LXNzZC1kNmRkYwoPCgZzb3VyY2USBWR1bW15EgcQ+IP4nJAyCtcCCjgKCF9fbmFtZV9fEixjb3J0ZXhfc2VsZl9tb25pdG9yaW5nX2Rpc2tfcmVhZF9ieXRlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKDQoGZGV2aWNlEgNzZGIKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAM5eKV1CEPKC+JyQMgqkAwo6CghfX25hbWVfXxIubG9nc19mb3J3YXJkZXJfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyCgoKAmxlEgQwLjI1Cg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoNCgVyb3V0ZRIEcm9vdAoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAAAoHgVBENqC+JyQMgqCBAouCghfX25hbWVfXxIia2VwbGVyX2NvbnRhaW5lcl9jb3JlX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAYWRhM2JiY2YyY2ZlN2VkODgzMDg2MmQ0Yjk5MDdmNTE3OGE4MzJhMGE2YzU1MjE3YWQ2ZGMzMzE4ODEzOGRmNQodCg5jb250YWluZXJfbmFtZRILZGlzdHJpYnV0b3IKIwoTY29udGFpbmVyX25hbWVzcGFjZRIMbWltaXItZGV2LTExCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDAoEbW9kZRIEaWRsZQoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgooCghwb2RfbmFtZRIcZGlzdHJpYnV0b3ItNjk2OTdmY2NjYi1xenpmYgodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAm0yHa+n3o8QBD4g/ickDIKkgQKMgoIX19uYW1lX18SJmtlcGxlcl9jb250YWluZXJfcGxhdGZvcm1fam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA5YWQ4MzQzOWI4OGI5ODVjODBjYjE3NTIzOTZkMjNkOWNlNDExNDJjZjA5NzJlMDQ4YTkwOTU5MzM4YjNlYjUzCiQKDmNvbnRhaW5lcl9uYW1lEhJuZXRmaWx0ZXItZXhwb3J0ZXIKKQoTY29udGFpbmVyX25hbWVzcGFjZRISbmV0ZmlsdGVyLWV4cG9ydGVyCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgokCghwb2RfbmFtZRIYbmV0ZmlsdGVyLWV4cG9ydGVyLWZ4dGg3Ch0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCQ4tsp3vB0FAEPiD+JyQMgrbAgouCghfX25hbWVfXxIiamFlZ2VyX3RyYWNlcl9yZXBvcnRlcl9zcGFuc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoMCgZyZXN1bHQSAm9rEhAJAAAAgJlUVkEQ2oL4nJAyCvADCjIKCF9fbmFtZV9fEiZrZXBsZXJfY29udGFpbmVyX3BsYXRmb3JtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAYjcyZjNkOTMzYjRlY2I3MzAzN2M5ZGY1ZTA3YWQ3YWYxZWIwOTNiOGJlZGMwOTE2ZjAyMGU1YTk1NjVjZGI0ZQoaCg5jb250YWluZXJfbmFtZRIIcHJvbXRhaWwKHgoTY29udGFpbmVyX25hbWVzcGFjZRIHZ2UtbG9ncwo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKGgoIcG9kX25hbWUSDnByb210YWlsLXIyNm01Ch0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCYXrUbiG3QRBEPiD+JyQMgrpAwo1CghfX25hbWVfXxIpa2VwbGVyX2NvbnRhaW5lcl9icGZfcGFnZV9jYWNoZV9oaXRfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDZhOGFjZTVkNjI4YTdiNWJhZGY2YmM4M2IzZTRlMzFlYjg5MmU2YjExMzllZDdmYTU2OTFkZWJhMGE3N2IwNzAKIQoOY29udGFpbmVyX25hbWUSD3F1ZXJ5LXNjaGVkdWxlcgojChNjb250YWluZXJfbmFtZXNwYWNlEgxtaW1pci1kZXYtMTEKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgorCghwb2RfbmFtZRIfcXVlcnktc2NoZWR1bGVyLTY4ZGM5Nzk0NC1jdHdnagoNCgZzb3VyY2USA2JwZhIHEPiD+JyQMgqrAwoyCghfX25hbWVfXxIma2VwbGVyX2NvbnRhaW5lcl9icGZfY3B1X3RpbWVfbXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgogCgxjb250YWluZXJfaWQSEHN5c3RlbV9wcm9jZXNzZXMKIgoOY29udGFpbmVyX25hbWUSEHN5c3RlbV9wcm9jZXNzZXMKHQoTY29udGFpbmVyX25hbWVzcGFjZRIGc3lzdGVtCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHAoIcG9kX25hbWUSEHN5c3RlbV9wcm9jZXNzZXMKDQoGc291cmNlEgNicGYSEAkAAAAAMUU8QRD4g/ickDIK8wMKMAoIX19uYW1lX18SJGtlcGxlcl9jb250YWluZXJfdW5jb3JlX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAMTQ1Njc3NTFiNTZkNTM5NTdmNmIyN2I2YTBhNTZmODgyOTFmMDllYzA2MTcxYTc3OGVkZGFkZDI2MTg5MjBhYgofCg5jb250YWluZXJfbmFtZRINaXAtbWFzcS1hZ2VudAoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHwoIcG9kX25hbWUSE2lwLW1hc3EtYWdlbnQtNzk3Y24KHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEgcQ+IP4nJAyCtIDCikKCF9fbmFtZV9fEh1rZXBsZXJfY29udGFpbmVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJANTZhYTVkOGY4ZjcxOTcxNWY2ZDRlMmYxOTkwYWExMmNiOTBiNmRiOGViN2Y5MjA4YjIyZGU0MDJlMGVmZTQwMgofCg5jb250YWluZXJfbmFtZRINbm9kZS1leHBvcnRlcgokChNjb250YWluZXJfbmFtZXNwYWNlEg1ub2RlLWV4cG9ydGVyCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgofCghwb2RfbmFtZRITbm9kZS1leHBvcnRlci02OWJ0YhIHEPiD+JyQMgr+AwouCghfX25hbWVfXxIia2VwbGVyX2NvbnRhaW5lcl9kcmFtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAZTQ2ODYzMmZmZDc0MjQ4ZDhjZTFmNzIyNGQyNWE2YzE4ZmJlYzRmMTRiMzkzMjcyNmFlYWI3MmYxYzhiODM0ZAodCg5jb250YWluZXJfbmFtZRILbWltaXItd3JpdGUKIwoTY29udGFpbmVyX25hbWVzcGFjZRIMbWltaXItZGV2LTA5CjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgohCghwb2RfbmFtZRIVbWltaXItd3JpdGUtem9uZS1iLTIwCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCR1aZDvfsMFAEPiD+JyQMgrIAgpCCghfX25hbWVfXxI2bG9raV9jYW5hcnlfc3BvdF9jaGVja19yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtsb2tpLWNhbmFyeQo2CghpbnN0YW5jZRIqbG9raS1jYW5hcnktaHhma2Q6bG9raS1jYW5hcnk6aHR0cC1tZXRyaWNzCh8KA2pvYhIYcHJvbXRhaWwtb3BzL2xva2ktY2FuYXJ5CgoKAmxlEgQxMC4wChkKCW5hbWVzcGFjZRIMcHJvbXRhaWwtb3BzChgKA3BvZBIRbG9raS1jYW5hcnktaHhma2QSEAkAAAAAgNDFQBCog/ickDIK8gIKLwoIX19uYW1lX18SI2NvcnRleF9yZXF1ZXN0X21lc3NhZ2VfYnl0ZXNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCgoCbGUSBCtJbmYKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwChYKBXJvdXRlEg1pbmdlc3Rlcl9yaW5nEhAJAAAAAAAA8D8Q8oL4nJAyCtQCCjUKCF9fbmFtZV9fEiljb3J0ZXhfZGlzdHJpYnV0b3JfcmVjZWl2ZWRfc2FtcGxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxNzg1NRIQCQAAAABuzGpBEPKC+JyQMgqdAwpCCghfX25hbWVfXxI2Y29ydGV4X2luZ2VzdGVyX2NsaWVudF9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCwoCbGUSBTAuMDE2ChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCikKCW9wZXJhdGlvbhIcL2dycGMuaGVhbHRoLnYxLkhlYWx0aC9DaGVjawojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKEQoLc3RhdHVzX2NvZGUSAk9LEhAJAAAAAFpCRkEQ8oL4nJAyCqsDCjIKCF9fbmFtZV9fEiZjb3J0ZXhfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgkKAmxlEgMyLjUKDgoGbWV0aG9kEgRnUlBDChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAomCgVyb3V0ZRIdL2Rpc3RyaWJ1dG9yLkRpc3RyaWJ1dG9yL1B1c2gKFwoLc3RhdHVzX2NvZGUSCEludGVybmFsCgsKAndzEgVmYWxzZRIQCQAAAAAAAPA/EPKC+JyQMgr6AwowCghfX25hbWVfXxIka2VwbGVyX2NvbnRhaW5lcl91bmNvcmVfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkBlMWMwMjhjMjdlMDk3MmFkNzkwNGQ5MzU1YzNlNTZhMWVlMWI5YjBjMzg4MmJiNDYzOGYwZTI4ZDA2ZDkzNDE2CiYKDmNvbnRhaW5lcl9uYW1lEhRjc2ktZHJpdmVyLXJlZ2lzdHJhcgoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHAoIcG9kX25hbWUSEHBkY3NpLW5vZGUtYmhzeGoKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEgcQ+IP4nJAyCqoDCkIKCF9fbmFtZV9fEjZjb3J0ZXhfaW5nZXN0ZXJfY2xpZW50X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoKCgJsZRIEK0luZgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwopCglvcGVyYXRpb24SHC9ncnBjLmhlYWx0aC52MS5IZWFsdGgvQ2hlY2sKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCh8KC3N0YXR1c19jb2RlEhBEZWFkbGluZUV4Y2VlZGVkEhAJAAAAAAAACEAQ8oL4nJAyCtsCCjQKCF9fbmFtZV9fEihjb3J0ZXhfZGlzdHJpYnV0b3JfZXhlbXBsYXJzX3Blcl9yZXF1ZXN0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTEwOTEzIhgIsKIBKZ2XjM8IulslMLCiAXjygvickDIKuQMKOQoIX19uYW1lX18SLXRoYW5vc19jYWNoZV9vcGVyYXRpb25fZGF0YV9zaXplX2J5dGVzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUyNTYuMAofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA3NldAojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAAAAAIBsQBDygvickDIKnQMKMgoIX19uYW1lX18SJmNvcnRleF9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCAoCbGUSAjEwCg4KBm1ldGhvZBIEZ1JQQwoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKHgoFcm91dGUSFS9odHRwZ3JwYy5IVFRQL0hhbmRsZQoSCgtzdGF0dXNfY29kZRIDNDAwCgsKAndzEgVmYWxzZRIQCQAAAACAesZAEPKC+JyQMgqeBAowCghfX25hbWVfXxIka2VwbGVyX2NvbnRhaW5lcl91bmNvcmVfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA3YzEwYTBlZmEzNDE0NDcwNjVmMmQ5NTIwYWVjODU5MGUyMzAwYzkyMjk3NWRiZDA3NDljNDE3YzBmZTg2OGJiChwKDmNvbnRhaW5lcl9uYW1lEgprdWJlLXByb3h5CiIKE2NvbnRhaW5lcl9uYW1lc3BhY2USC2t1YmUtc3lzdGVtCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgpKCghwb2RfbmFtZRI+a3ViZS1wcm94eS1na2UtZGV2LXVzLWNlbnRyYWwtMC1tYWluLW4yczE2LTMtMWRkLTliNTAyZDk2LXR2bWsKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEgcQ+IP4nJAyCpMDCjIKCF9fbmFtZV9fEiZjb3J0ZXhfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgkKAmxlEgMxMDAKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwChQKBXJvdXRlEgtkZWJ1Z19wcHJvZgoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAADQO/NAEPKC+JyQMgqMAwoyCghfX25hbWVfXxImY29ydGV4X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoICgJsZRICMTAKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg4KBXJvdXRlEgVyZWFkeQoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAABA3uxAEPKC+JyQMgrUAgo1CghfX25hbWVfXxIpY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX3NhbXBsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTY1NDMSEAkAAABwOKZ+QRDygvickDIKxgIKJwoIX19uYW1lX18SG2dvX21lbXN0YXRzX21zcGFuX3N5c19ieXRlcwofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZhIQCQAAAAAA5ilBENqC+JyQMgq+Ago3CghfX25hbWVfXxIrbG9raV9jYW5hcnlfcmVzcG9uc2VfbGF0ZW5jeV9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILbG9raS1jYW5hcnkKNgoIaW5zdGFuY2USKmxva2ktY2FuYXJ5LWh4ZmtkOmxva2ktY2FuYXJ5Omh0dHAtbWV0cmljcwofCgNqb2ISGHByb210YWlsLW9wcy9sb2tpLWNhbmFyeQoLCgJsZRIFMjU2LjAKGQoJbmFtZXNwYWNlEgxwcm9tdGFpbC1vcHMKGAoDcG9kEhFsb2tpLWNhbmFyeS1oeGZrZBIQCQAAAACQzPdAEKiD+JyQMgqJAwoxCghfX25hbWVfXxIlY29ydGV4X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19jb3VudAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoWCgVyb3V0ZRINaW5nZXN0ZXJfcmluZwoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAAAAAPA/EPKC+JyQMgqOBAouCghfX25hbWVfXxIia2VwbGVyX2NvbnRhaW5lcl9kcmFtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAOWFkODM0MzliODhiOTg1YzgwY2IxNzUyMzk2ZDIzZDljZTQxMTQyY2YwOTcyZTA0OGE5MDk1OTMzOGIzZWI1MwokCg5jb250YWluZXJfbmFtZRISbmV0ZmlsdGVyLWV4cG9ydGVyCikKE2NvbnRhaW5lcl9uYW1lc3BhY2USEm5ldGZpbHRlci1leHBvcnRlcgo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKJAoIcG9kX25hbWUSGG5ldGZpbHRlci1leHBvcnRlci1meHRoNwodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAnRItv5fmoZQBD4g/ickDIKxwIKQgoIX19uYW1lX18SNmxva2lfY2FuYXJ5X3Nwb3RfY2hlY2tfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILbG9raS1jYW5hcnkKNgoIaW5zdGFuY2USKmxva2ktY2FuYXJ5LWh4ZmtkOmxva2ktY2FuYXJ5Omh0dHAtbWV0cmljcwofCgNqb2ISGHByb210YWlsLW9wcy9sb2tpLWNhbmFyeQoJCgJsZRIDMC4xChkKCW5hbWVzcGFjZRIMcHJvbXRhaWwtb3BzChgKA3BvZBIRbG9raS1jYW5hcnktaHhma2QSEAkAAAAAAEPFQBCog/ickDIK/wIKKgoIX19uYW1lX18SHnByb210YWlsX211dGF0ZWRfZW50cmllc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChUKCWNvbnRhaW5lchIIcHJvbXRhaWwKMAoEaG9zdBIoY29ydGV4LWd3Lmxva2ktZGV2LXNzZC5zdmMuY2x1c3Rlci5sb2NhbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKGQoJbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKIgoDcG9kEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42enoKGAoGcmVhc29uEg5pbmdlc3Rlcl9lcnJvchIHEOKD+JyQMgrcAgomCghfX25hbWVfXxIaamFlZ2VyX3RyYWNlcl90cmFjZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAobCgljb250YWluZXISDmxvZ3MtZm9yd2FyZGVyCkcKCGluc3RhbmNlEjtsb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmOmxvZ3MtZm9yd2FyZGVyOmh0dHAtbWV0cmljcwoiCgNqb2ISG2luc2lnaHQtbG9ncy9sb2dzLWZvcndhcmRlcgoZCgluYW1lc3BhY2USDGluc2lnaHQtbG9ncwomCgNwb2QSH2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2YKDAoHc2FtcGxlZBIBeQoQCgVzdGF0ZRIHc3RhcnRlZBIHENqC+JyQMgriAgpDCghfX25hbWVfXxI3Y29ydGV4X2Rpc3RyaWJ1dG9yX2xhdGVzdF9zZWVuX3NhbXBsZV90aW1lc3RhbXBfc2Vjb25kcwofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxMzY3NxIQCcP1+Oc2qtlBEPKC+JyQMgr9AwouCghfX25hbWVfXxIia2VwbGVyX2NvbnRhaW5lcl9jb3JlX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAMTQ1Njc3NTFiNTZkNTM5NTdmNmIyN2I2YTBhNTZmODgyOTFmMDllYzA2MTcxYTc3OGVkZGFkZDI2MTg5MjBhYgofCg5jb250YWluZXJfbmFtZRINaXAtbWFzcS1hZ2VudAoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHwoIcG9kX25hbWUSE2lwLW1hc3EtYWdlbnQtNzk3Y24KHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJmG4Sg8AyYUAQ+IP4nJAyCooDCisKCF9fbmFtZV9fEh9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKFQoJY29udGFpbmVyEghwcm9tdGFpbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKCQoCbGUSAzEuMAoNCgZtZXRob2QSA0dFVAoZCgluYW1lc3BhY2USDGxva2ktZGV2LXNzZAoiCgNwb2QSG3Byb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6egoQCgVyb3V0ZRIHbWV0cmljcwoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAAAAmqRAEOKD+JyQMgq5AwoxCghfX25hbWVfXxIldGhhbm9zX2NhY2hlX29wZXJhdGlvbl9mYWlsdXJlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCh8KBG5hbWUSF2FnZ3JlZ2F0aW9uLXJ1bGVzLWNhY2hlChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChUKCW9wZXJhdGlvbhIIZmx1c2hhbGwKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwChcKBnJlYXNvbhINbmV0d29yay1lcnJvchIHEPKC+JyQMgqJAworCghfX25hbWVfXxIfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChUKCWNvbnRhaW5lchIIcHJvbXRhaWwKPQoIaW5zdGFuY2USMXByb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ejpwcm9tdGFpbDpodHRwLW1ldHJpY3MKKQoDam9iEiJsb2tpLWRldi1zc2QvcHJvbXRhaWwtbG9raS1kZXYtc3NkCgoKAmxlEgQwLjA1Cg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMbG9raS1kZXYtc3NkCiIKA3BvZBIbcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6Cg4KBXJvdXRlEgVyZWFkeQoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIQCQAAAAAA7K5AEOKD+JyQMgr0AwouCghfX25hbWVfXxIia2VwbGVyX2NvbnRhaW5lcl9kcmFtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJANDA2NmU1MWMzYjE4NTM5MmMyMzZiMjE0N2E2YWE5ZWU0Mjg2MzJiZmQ5YjkyMTVlZWM0N2ExODY4NWJhY2M0OAoXCg5jb250YWluZXJfbmFtZRIFYmV5bGEKJAoTY29udGFpbmVyX25hbWVzcGFjZRINYmV5bGEtbmV0d29yawo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHwoIcG9kX25hbWUSE2JleWxhLW5ldHdvcmstNzR0eGMKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJMzMzMzMzFkAQ+IP4nJAyCvoCCjIKCF9fbmFtZV9fEiZjb3J0ZXhfZGlzdHJpYnV0b3Jfc2FtcGxlc19wZXJfcmVxdWVzdAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxOTI1OSI5CNFNGQAAAID3pWNBKZ2XjM8IulslMMEBWgIQDGIUkANE0gJz/wFB1AEhwAGeCMIN9it48oL4nJAyCskDCikKCF9fbmFtZV9fEh1rZXBsZXJfY29udGFpbmVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAOWI3MTdjOWZhYmY2Mzk0ZGJiMjRhN2FhMzZmZGNiMGM5ZDJhMjQwYmUxZjgzN2VkMWViNDY1MzFjNzNiMjA2OAodCg5jb250YWluZXJfbmFtZRILaW5zdGFsbC1jbmkKIgoTY29udGFpbmVyX25hbWVzcGFjZRILa3ViZS1zeXN0ZW0KNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh0KCHBvZF9uYW1lEhFjYWxpY28tbm9kZS14ZmZieBIHEPiD+JyQMgqdAwovCghfX25hbWVfXxIjY29ydGV4X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19zdW0KHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoOCgZtZXRob2QSBGdSUEMKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCiYKBXJvdXRlEh0vZGlzdHJpYnV0b3IuRGlzdHJpYnV0b3IvUHVzaAoXCgtzdGF0dXNfY29kZRIISW50ZXJuYWwKCwoCd3MSBWZhbHNlEhAJtVlQc00EAEAQ8oL4nJAyCpwDCkIKCF9fbmFtZV9fEjZjb3J0ZXhfaW5nZXN0ZXJfY2xpZW50X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoLCgJsZRIFMS4wMjQKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIgoJb3BlcmF0aW9uEhUvY29ydGV4LkluZ2VzdGVyL1B1c2gKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwChcKC3N0YXR1c19jb2RlEghDYW5jZWxlZBIQCQAAAAAAgEFAEPKC+JyQMgraAgo7CghfX25hbWVfXxIvY29ydGV4X3NlbGZfbW9uaXRvcmluZ19kaXNrX3dyaXR0ZW5fYnl0ZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCg0KBmRldmljZRIDc2RmCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAAAAfuPwQRDygvickDIKvAMKNQoIX19uYW1lX18SKWNvcnRleF9rdl9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKIQoHa3ZfbmFtZRIWZGlzdHJpYnV0b3ItbGlmZWN5Y2xlcgoHCgJsZRIBNQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA0NBUwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDwoEcm9sZRIHcHJpbWFyeQoSCgtzdGF0dXNfY29kZRIDMjAwChIKBHR5cGUSCm1lbWJlcmxpc3QSEAkAAAAAAD/DQBDygvickDIK8AMKLgoIX19uYW1lX18SImtlcGxlcl9jb250YWluZXJfY29yZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDhiN2M0MDNhNmRhZDJhZTkwNTY1OTMyM2Q0MmUzZWE0ODEyYjZiNzU5NTUxZDFhYjIwOWZhMjBmZjdkYTQ3NWUKIQoOY29udGFpbmVyX25hbWUSD2tlcGxlci1leHBvcnRlcgodChNjb250YWluZXJfbmFtZXNwYWNlEgZrZXBsZXIKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyChgKCHBvZF9uYW1lEgxrZXBsZXItdmRqdzIKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJRrbz/dT4K0AQ+IP4nJAyCtUCCjYKCF9fbmFtZV9fEipjb3J0ZXhfZGlzdHJpYnV0b3JfcmVjZWl2ZWRfcmVxdWVzdHNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTA5NTkSEAkAAAAAAAAIQBDygvickDIK+wMKLgoIX19uYW1lX18SImtlcGxlcl9jb250YWluZXJfY29yZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDY3Y2I4OTYzYTMyNjE1NTg3ZmI5NWRlM2FmZjQwNTBkOGY5NjkxOTRlM2FkOWE0ODkyZjg5Y2Y2ZTM2MTU5YzMKHAoOY29udGFpbmVyX25hbWUSCm5vZGUtY2FjaGUKIgoTY29udGFpbmVyX25hbWVzcGFjZRILa3ViZS1zeXN0ZW0KNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCiAKCHBvZF9uYW1lEhRub2RlLWxvY2FsLWRucy13anpobAodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAkZBFYOLdI8QBD4g/ickDIKiAMKOgoIX19uYW1lX18SLmdyYWZhbmFjb21fYXBpX3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoLCgRjb2RlEgM1MDIKGwoJY29udGFpbmVyEg5sb2dzLWZvcndhcmRlcgpHCghpbnN0YW5jZRI7bG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZjpsb2dzLWZvcndhcmRlcjpodHRwLW1ldHJpY3MKIgoDam9iEhtpbnNpZ2h0LWxvZ3MvbG9ncy1mb3J3YXJkZXIKCQoCbGUSAzAuNQoZCgluYW1lc3BhY2USDGluc2lnaHQtbG9ncwoVCgRwYXRoEg1ob3N0ZWQtYWxlcnRzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZhIQCQAAAAAAAPA/ENqC+JyQMgrMAgoqCghfX25hbWVfXxIea2Fma2Ffd3JpdGVyX2Rpc2Nvbm5lY3RzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKEAoHbm9kZV9pZBIFMjAwMDAKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAACEtUAQ8oL4nJAyCrQDCjIKCF9fbmFtZV9fEiZjb3J0ZXhfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUwLjAyNQoOCgZtZXRob2QSBGdSUEMKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCiYKBXJvdXRlEh0vZGlzdHJpYnV0b3IuRGlzdHJpYnV0b3IvUHVzaAoeCgtzdGF0dXNfY29kZRIPSW52YWxpZEFyZ3VtZW50CgsKAndzEgVmYWxzZRIQCQAAAAAAABxAEPKC+JyQMgqtAgobCghfX25hbWVfXxIPcHJvY2Vzc19tYXhfZmRzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKFQoJY29udGFpbmVyEghwcm9tdGFpbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKGQoJbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKIgoDcG9kEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42enoSEAkAAAAAAAAwQRDig/ickDIKygIKLgoIX19uYW1lX18SImphZWdlcl90cmFjZXJfcmVwb3J0ZXJfc3BhbnNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoVCgljb250YWluZXISCHByb210YWlsCj0KCGluc3RhbmNlEjFwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42eno6cHJvbXRhaWw6aHR0cC1tZXRyaWNzCikKA2pvYhIibG9raS1kZXYtc3NkL3Byb210YWlsLWxva2ktZGV2LXNzZAoZCgluYW1lc3BhY2USDGxva2ktZGV2LXNzZAoiCgNwb2QSG3Byb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6egoRCgZyZXN1bHQSB2Ryb3BwZWQSBxDig/ickDIKxAIKLgoIX19uYW1lX18SImNvcnRleF9zZWxmX21vbml0b3JpbmdfZGlza19pb19ub3cKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCg0KBmRldmljZRIDc2RlCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASBxDygvickDIKggQKMgoIX19uYW1lX18SJmtlcGxlcl9jb250YWluZXJfcGxhdGZvcm1fam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA5ZjI2ZmMzNTRhMThlY2U0Njk3MGY1ZjMxZjFmNGJhOWVjNGZhMTg4OTRiNGZmMmYwYWQzMzExZThiMTA4YzI3ChoKDmNvbnRhaW5lcl9uYW1lEghwcm9tdGFpbAojChNjb250YWluZXJfbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCicKCHBvZF9uYW1lEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtZDZkZGMKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJDAIrB2kbEUEQ+IP4nJAyCoMECi8KCF9fbmFtZV9fEiNrZXBsZXJfY29udGFpbmVyX290aGVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAYWRhM2JiY2YyY2ZlN2VkODgzMDg2MmQ0Yjk5MDdmNTE3OGE4MzJhMGE2YzU1MjE3YWQ2ZGMzMzE4ODEzOGRmNQodCg5jb250YWluZXJfbmFtZRILZGlzdHJpYnV0b3IKIwoTY29udGFpbmVyX25hbWVzcGFjZRIMbWltaXItZGV2LTExCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDAoEbW9kZRIEaWRsZQoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgooCghwb2RfbmFtZRIcZGlzdHJpYnV0b3ItNjk2OTdmY2NjYi1xenpmYgodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAkUrkfheiRFQBD4g/ickDIK8AIKVAoIX19uYW1lX18SSGNvcnRleF9oYV90cmFja2VyX2VsZWN0ZWRfcmVwbGljYV9jaGFuZ2VfcHJvcGFnYXRpb25fdGltZV9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgoKAmxlEgQrSW5mChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAABAeP9AEPKC+JyQMgrPAgowCghfX25hbWVfXxIkY29ydGV4X2Rpc3RyaWJ1dG9yX3JlcXVlc3RzX2luX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTE3ODU1EhAJAAAAAKAT4EAQ8oL4nJAyCpgDCikKCF9fbmFtZV9fEh1rZXBsZXJfbm9kZV9kcmFtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyCkgKEWV4cG9ydGVkX2luc3RhbmNlEjNna2UtZGV2LXVzLWNlbnRyYWwtMC1tYWluLW4yczE2LTMtMWRkLTliNTAyZDk2LXR2bWsKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyCgwKB3BhY2thZ2USATAKEwoDcG9kEgxrZXBsZXItdmRqdzIKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJYOXQItPO7UAQ+IP4nJAyCrwCCjcKCF9fbmFtZV9fEitsb2tpX2NhbmFyeV9yZXNwb25zZV9sYXRlbmN5X3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtsb2tpLWNhbmFyeQo2CghpbnN0YW5jZRIqbG9raS1jYW5hcnktaHhma2Q6bG9raS1jYW5hcnk6aHR0cC1tZXRyaWNzCh8KA2pvYhIYcHJvbXRhaWwtb3BzL2xva2ktY2FuYXJ5CgkKAmxlEgMyLjAKGQoJbmFtZXNwYWNlEgxwcm9tdGFpbC1vcHMKGAoDcG9kEhFsb2tpLWNhbmFyeS1oeGZrZBIQCQAAAABgp/dAEKiD+JyQMgrUAgo1CghfX25hbWVfXxIpY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX3NhbXBsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTM2NzcSEAkAAAAAACjDQBDygvickDIK1QIKPgoIX19uYW1lX18SMmNvcnRleF9zZWxmX21vbml0b3JpbmdfZGlza193cml0ZXNfY29tcGxldGVkX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgoOCgZkZXZpY2USBGRtLTAKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIHEPKC+JyQMgq0AwoxCghfX25hbWVfXxIldGhhbm9zX2NhY2hlX29wZXJhdGlvbl9mYWlsdXJlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCh8KBG5hbWUSF2FnZ3JlZ2F0aW9uLXJ1bGVzLWNhY2hlChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChAKCW9wZXJhdGlvbhIDc2V0CiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoXCgZyZWFzb24SDW5ldHdvcmstZXJyb3ISBxDygvickDIKjgMKMgoIX19uYW1lX18SJmNvcnRleF9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCwoCbGUSBTAuMDA1Cg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoWCgVyb3V0ZRINaW5nZXN0ZXJfcmluZwoSCgtzdGF0dXNfY29kZRIDMjAwCgsKAndzEgVmYWxzZRIHEPKC+JyQMgqBBAoyCghfX25hbWVfXxIma2VwbGVyX2NvbnRhaW5lcl9wbGF0Zm9ybV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDE0NTY3NzUxYjU2ZDUzOTU3ZjZiMjdiNmEwYTU2Zjg4MjkxZjA5ZWMwNjE3MWE3NzhlZGRhZGQyNjE4OTIwYWIKHwoOY29udGFpbmVyX25hbWUSDWlwLW1hc3EtYWdlbnQKIgoTY29udGFpbmVyX25hbWVzcGFjZRILa3ViZS1zeXN0ZW0KNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh8KCHBvZF9uYW1lEhNpcC1tYXNxLWFnZW50LTc5N2NuCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCUa28/3UoGZAEPiD+JyQMgrFAgo+CghfX25hbWVfXxIybWVtYmVybGlzdF90Y3BfdHJhbnNwb3J0X3Vua25vd25fY29ubmVjdGlvbnNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASBxDygvickDIK5wIKMgoIX19uYW1lX18SJmNvcnRleF9kaXN0cmlidXRvcl9zYW1wbGVzX3Blcl9yZXF1ZXN0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTE5MDU1IiYIARkAAAAAAOByQCmdl4zPCLpbJTAAWgQIEhABYgECePKC+JyQMgrJAgozCghfX25hbWVfXxInamFlZ2VyX3RyYWNlcl9iYWdnYWdlX3RydW5jYXRpb25zX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGwoJY29udGFpbmVyEg5sb2dzLWZvcndhcmRlcgpHCghpbnN0YW5jZRI7bG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZjpsb2dzLWZvcndhcmRlcjpodHRwLW1ldHJpY3MKIgoDam9iEhtpbnNpZ2h0LWxvZ3MvbG9ncy1mb3J3YXJkZXIKGQoJbmFtZXNwYWNlEgxpbnNpZ2h0LWxvZ3MKJgoDcG9kEh9sb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmEgcQ2oL4nJAyCvUDCi4KCF9fbmFtZV9fEiJrZXBsZXJfY29udGFpbmVyX2NvcmVfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA3ZmU5YTA3MTY2ZmQ3OGFiOTY3MTY5YzMyNDk0ZjgzMzUxNjVkZTFkZTk0NWViMWRhZjNmOGM4MjVkYTdkY2RjChoKDmNvbnRhaW5lcl9uYW1lEghpbmdlc3RlcgojChNjb250YWluZXJfbmFtZXNwYWNlEgxtaW1pci1kZXYtMTQKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh4KCHBvZF9uYW1lEhJpbmdlc3Rlci16b25lLWItMTEKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJMzMzM5827EAQ+IP4nJAyCpcDCjQKCF9fbmFtZV9fEihwcm9tdGFpbF9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKFQoJY29udGFpbmVyEghwcm9tdGFpbAowCgRob3N0Eihjb3J0ZXgtZ3cubG9raS1kZXYtc3NkLnN2Yy5jbHVzdGVyLmxvY2FsCj0KCGluc3RhbmNlEjFwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42eno6cHJvbXRhaWw6aHR0cC1tZXRyaWNzCikKA2pvYhIibG9raS1kZXYtc3NkL3Byb210YWlsLWxva2ktZGV2LXNzZAoJCgJsZRIDMC4xChkKCW5hbWVzcGFjZRIMbG9raS1kZXYtc3NkCiIKA3BvZBIbcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6ChIKC3N0YXR1c19jb2RlEgMyMDQSEAkAAAAAAADwPxDig/ickDIK7wIKKQoIX19uYW1lX18SHXJlc3BvbnNlX21lc3NhZ2VfYnl0ZXNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKFQoJY29udGFpbmVyEghwcm9tdGFpbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKEQoCbGUSCzIuNjIxNDRlKzA2Cg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMbG9raS1kZXYtc3NkCiIKA3BvZBIbcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6ChAKBXJvdXRlEgdtZXRyaWNzEhAJAAAAAACapEAQ4oP4nJAyCu0CCjIKCF9fbmFtZV9fEiZjb3J0ZXhfZGlzdHJpYnV0b3Jfc2FtcGxlc19wZXJfcmVxdWVzdAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxMDgwOSIsCGEZAAAAACC8/0ApnZeMzwi6WyUwBFoECAwQBmIHFAYXAQCKAXjygvickDIK7wIKVAoIX19uYW1lX18SSGNvcnRleF9oYV90cmFja2VyX2VsZWN0ZWRfcmVwbGljYV9jaGFuZ2VfcHJvcGFnYXRpb25fdGltZV9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgkKAmxlEgMxLjAKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAEB4/0AQ8oL4nJAyCoEECikKCF9fbmFtZV9fEh1rZXBsZXJfY29udGFpbmVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAN2MxMGEwZWZhMzQxNDQ3MDY1ZjJkOTUyMGFlYzg1OTBlMjMwMGM5MjI5NzVkYmQwNzQ5YzQxN2MwZmU4NjhiYgocCg5jb250YWluZXJfbmFtZRIKa3ViZS1wcm94eQoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKSgoIcG9kX25hbWUSPmt1YmUtcHJveHktZ2tlLWRldi11cy1jZW50cmFsLTAtbWFpbi1uMnMxNi0zLTFkZC05YjUwMmQ5Ni10dm1rEhAJEFg5tEhRo0AQ+IP4nJAyCv4DCi8KCF9fbmFtZV9fEiNrZXBsZXJfY29udGFpbmVyX290aGVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAMTQ1Njc3NTFiNTZkNTM5NTdmNmIyN2I2YTBhNTZmODgyOTFmMDllYzA2MTcxYTc3OGVkZGFkZDI2MTg5MjBhYgofCg5jb250YWluZXJfbmFtZRINaXAtbWFzcS1hZ2VudAoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHwoIcG9kX25hbWUSE2lwLW1hc3EtYWdlbnQtNzk3Y24KHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJc2iR7Xx/MkAQ+IP4nJAyCo8DCjIKCF9fbmFtZV9fEiZjb3J0ZXhfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUwLjAyNQoNCgZtZXRob2QSA0dFVAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDgoFcm91dGUSBXJlYWR5ChIKC3N0YXR1c19jb2RlEgMyMDAKCwoCd3MSBWZhbHNlEhAJAAAAAEDe7EAQ8oL4nJAyCuICCkMKCF9fbmFtZV9fEjdjb3J0ZXhfZGlzdHJpYnV0b3JfbGF0ZXN0X3NlZW5fc2FtcGxlX3RpbWVzdGFtcF9zZWNvbmRzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTE5MDU1EhAJdZMImjaq2UEQ8oL4nJAyCogDCjoKCF9fbmFtZV9fEi5ncmFmYW5hY29tX2FwaV9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKCwoEY29kZRIDNTAyChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyCgkKAmxlEgMyLjUKGQoJbmFtZXNwYWNlEgxpbnNpZ2h0LWxvZ3MKFQoEcGF0aBINaG9zdGVkLWFsZXJ0cwomCgNwb2QSH2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2YSEAkAAAAAAADwPxDagvickDIK1wIKMAoIX19uYW1lX18SJGNvcnRleF9kaXN0cmlidXRvcl9yZXF1ZXN0c19pbl90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoVCgR1c2VyEg1iaWxsaW5nLXN1cGVyEhAJAAAAAAAb5EAQ8oL4nJAyCtUCCjYKCF9fbmFtZV9fEipjb3J0ZXhfZGlzdHJpYnV0b3JfcmVjZWl2ZWRfcmVxdWVzdHNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTA4MDkSEAkAAAAAAEBYQBDygvickDIKugIKKAoIX19uYW1lX18SHGdvX21lbXN0YXRzX2hlYXBfaW51c2VfYnl0ZXMKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoVCgljb250YWluZXISCHByb210YWlsCj0KCGluc3RhbmNlEjFwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42eno6cHJvbXRhaWw6aHR0cC1tZXRyaWNzCikKA2pvYhIibG9raS1kZXYtc3NkL3Byb210YWlsLWxva2ktZGV2LXNzZAoZCgluYW1lc3BhY2USDGxva2ktZGV2LXNzZAoiCgNwb2QSG3Byb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ehIQCQAAAAAAqH1BEOKD+JyQMgqyAgoiCghfX25hbWVfXxIWc2NyYXBlX3NhbXBsZXNfc2NyYXBlZAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAAAAVJ1AEPKC+JyQMgrBAwo5CghfX25hbWVfXxItdGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kYXRhX3NpemVfYnl0ZXNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChQKB2JhY2tlbmQSCW1lbWNhY2hlZAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbXBvbmVudBILZGlzdHJpYnV0b3IKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKFAoCbGUSDjIuNjg0MzU0NTZlKzA4Ch8KBG5hbWUSF2FnZ3JlZ2F0aW9uLXJ1bGVzLWNhY2hlChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChgKCW9wZXJhdGlvbhILY29tcGFyZXN3YXAKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEgcQ8oL4nJAyCuYDCjAKCF9fbmFtZV9fEiRrZXBsZXJfY29udGFpbmVyX2JwZl9ibG9ja19pcnFfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQGVhODY4YzZhMTAwNGRkNTY0YzVkYjEwZTFlZWNjZTEwZTM0MDdiZTlmNjk4NTM5MjEyY2FhM2VhNjc5N2FjM2QKJAoOY29udGFpbmVyX25hbWUSEmNvbm50cmFjay1leHBvcnRlcgopChNjb250YWluZXJfbmFtZXNwYWNlEhJjb25udHJhY2stZXhwb3J0ZXIKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgokCghwb2RfbmFtZRIYY29ubnRyYWNrLWV4cG9ydGVyLXM5Z2w2Cg0KBnNvdXJjZRIDYnBmEgcQ+IP4nJAyCqMDCkIKCF9fbmFtZV9fEjZjb3J0ZXhfaW5nZXN0ZXJfY2xpZW50X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoLCgJsZRIFMC4wMDEKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIgoJb3BlcmF0aW9uEhUvY29ydGV4LkluZ2VzdGVyL1B1c2gKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCh4KC3N0YXR1c19jb2RlEg9JbnZhbGlkQXJndW1lbnQSEAkAAAAAAAAiQBDygvickDIKuQMKOgoIX19uYW1lX18SLnRoYW5vc19jYWNoZV9vcGVyYXRpb25fZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKFAoHYmFja2VuZBIJbWVtY2FjaGVkChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29tcG9uZW50EgtkaXN0cmlidXRvcgoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoKCgJsZRIEMC4wNQofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA3NldAojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAAAAAIBsQBDygvickDIKuwIKKQoIX19uYW1lX18SHXByb210YWlsX3RhcmdldHNfYWN0aXZlX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKFQoJY29udGFpbmVyEghwcm9tdGFpbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKGQoJbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKIgoDcG9kEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42enoSEAkAAAAAAADwPxDig/ickDIKzQIKLwoIX19uYW1lX18SI2phZWdlcl90cmFjZXJfc2FtcGxlcl9xdWVyaWVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCgwKBnJlc3VsdBICb2sSEAkAAAAAgD7DQBDygvickDIK9gMKLQoIX19uYW1lX18SIWtlcGxlcl9jb250YWluZXJfZ3B1X2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAOWFkODM0MzliODhiOTg1YzgwY2IxNzUyMzk2ZDIzZDljZTQxMTQyY2YwOTcyZTA0OGE5MDk1OTMzOGIzZWI1MwokCg5jb250YWluZXJfbmFtZRISbmV0ZmlsdGVyLWV4cG9ydGVyCikKE2NvbnRhaW5lcl9uYW1lc3BhY2USEm5ldGZpbHRlci1leHBvcnRlcgo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKJAoIcG9kX25hbWUSGG5ldGZpbHRlci1leHBvcnRlci1meHRoNwoPCgZzb3VyY2USBWR1bW15EgcQ+IP4nJAyCvIDCjAKCF9fbmFtZV9fEiRrZXBsZXJfY29udGFpbmVyX3VuY29yZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDE1NDgxMjM5OTJkNzBhMzExYTVlZTlhYzBjZDBjYTdjZjZjMzBhMjZhNDI3NTI1MTZkYzlkMDdmYjM3ZTY0ODUKGgoOY29udGFpbmVyX25hbWUSCGluZ2VzdGVyCiMKE2NvbnRhaW5lcl9uYW1lc3BhY2USDG1pbWlyLWRldi0xMQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHwoIcG9kX25hbWUSE2luZ2VzdGVyLXpvbmUtYy0xMDEKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEgcQ+IP4nJAyCt0DCi0KCF9fbmFtZV9fEiFrZXBsZXJfY29udGFpbmVyX2dwdV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDdmZTlhMDcxNjZmZDc4YWI5NjcxNjljMzI0OTRmODMzNTE2NWRlMWRlOTQ1ZWIxZGFmM2Y4YzgyNWRhN2RjZGMKGgoOY29udGFpbmVyX25hbWUSCGluZ2VzdGVyCiMKE2NvbnRhaW5lcl9uYW1lc3BhY2USDG1pbWlyLWRldi0xNAo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHgoIcG9kX25hbWUSEmluZ2VzdGVyLXpvbmUtYi0xMQoPCgZzb3VyY2USBWR1bW15EgcQ+IP4nJAyCoQDCjAKCF9fbmFtZV9fEiRjb3J0ZXhfcmVzcG9uc2VfbWVzc2FnZV9ieXRlc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoSCgJsZRIMMS4wNDg1NzZlKzA2Cg4KBm1ldGhvZBIEZ1JQQwoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKHgoFcm91dGUSFS9odHRwZ3JwYy5IVFRQL0hhbmRsZRIQCQAAAMBiFV1BEPKC+JyQMgrAAwo1CghfX25hbWVfXxIpY29ydGV4X2t2X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgohCgdrdl9uYW1lEhZkaXN0cmlidXRvci1saWZlY3ljbGVyCgsKAmxlEgUwLjAyNQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA0NBUwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDwoEcm9sZRIHcHJpbWFyeQoSCgtzdGF0dXNfY29kZRIDMjAwChIKBHR5cGUSCm1lbWJlcmxpc3QSEAkAAAAAAD/DQBDygvickDIKzAIKNgoIX19uYW1lX18SKmNvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9tZXRhZGF0YV90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxNDAyNxIHEPKC+JyQMgrXAgo6CghfX25hbWVfXxIuY29ydGV4X2Rpc3RyaWJ1dG9yX3NhbXBsZV9kZWxheV9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUxMjAuMAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAIATtmMOQhDygvickDIK4QIKKgoIX19uYW1lX18SHmNvcnRleF9kaXNjYXJkZWRfc2FtcGxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoWCgZyZWFzb24SDHJhdGVfbGltaXRlZAoNCgR1c2VyEgUxNjU0MxIQCQAAAAAAiMNAEPKC+JyQMgrXAgo0CghfX25hbWVfXxIoY29ydGV4X2Rpc3RyaWJ1dG9yX2V4ZW1wbGFyc19wZXJfcmVxdWVzdAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxMDk1OSIUCAMpnZeMzwi6WyUwA3jygvickDIK7wMKMQoIX19uYW1lX18SJWtlcGxlcl9jb250YWluZXJfcGFja2FnZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQGI3MmYzZDkzM2I0ZWNiNzMwMzdjOWRmNWUwN2FkN2FmMWViMDkzYjhiZWRjMDkxNmYwMjBlNWE5NTY1Y2RiNGUKGgoOY29udGFpbmVyX25hbWUSCHByb210YWlsCh4KE2NvbnRhaW5lcl9uYW1lc3BhY2USB2dlLWxvZ3MKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyChoKCHBvZF9uYW1lEg5wcm9tdGFpbC1yMjZtNQodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAklBoGV32rqQBD4g/ickDIK4gIKMgoIX19uYW1lX18SJmthZmthX3dyaXRlcl93cml0ZV93YWl0X3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCwoCbGUSBTEuMDI0ChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChEKB25vZGVfaWQSBnNlZWRfMAojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAAAAAAAAQBDygvickDIK1QIKNgoIX19uYW1lX18SKmNvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9yZXF1ZXN0c190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxNjU0MxIQCQAAAACA0MNAEPKC+JyQMgrOAgovCghfX25hbWVfXxIjY29ydGV4X2Rpc3RyaWJ1dG9yX3NhbXBsZXNfaW5fdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTU2MjkSEAkAAAAAIObrQBDygvickDIKywMKMQoIX19uYW1lX18SJWtlcGxlcl9jb250YWluZXJfcGFja2FnZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgogCgxjb250YWluZXJfaWQSEHN5c3RlbV9wcm9jZXNzZXMKIgoOY29udGFpbmVyX25hbWUSEHN5c3RlbV9wcm9jZXNzZXMKHQoTY29udGFpbmVyX25hbWVzcGFjZRIGc3lzdGVtCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgocCghwb2RfbmFtZRIQc3lzdGVtX3Byb2Nlc3NlcwodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAmmm8QgsHG7QBD4g/ickDIKlAMKNwoIX19uYW1lX18SK2xvZ3NfZm9yd2FyZGVyX3JlcXVlc3RfbWVzc2FnZV9ieXRlc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAobCgljb250YWluZXISDmxvZ3MtZm9yd2FyZGVyCkcKCGluc3RhbmNlEjtsb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmOmxvZ3MtZm9yd2FyZGVyOmh0dHAtbWV0cmljcwoiCgNqb2ISG2luc2lnaHQtbG9ncy9sb2dzLWZvcndhcmRlcgoRCgJsZRILMi42MjE0NGUrMDcKDgoGbWV0aG9kEgRQT1NUChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoZCgVyb3V0ZRIQbG9raV9hcGlfdjFfcHVzaBIQCQAAAOAFoolBENqC+JyQMgqrAwo0CghfX25hbWVfXxIoY29ydGV4X2t2X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19jb3VudAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCiAKB2t2X25hbWUSFWRpc3RyaWJ1dG9yLWhhdHJhY2tlcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA0NBUwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDwoEcm9sZRIHcHJpbWFyeQoSCgtzdGF0dXNfY29kZRIDMjAwCgwKBHR5cGUSBGV0Y2QSEAkAAAAAAFKnQBDygvickDIK/QIKKAoIX19uYW1lX18SHHByb210YWlsX2Ryb3BwZWRfYnl0ZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoVCgljb250YWluZXISCHByb210YWlsCjAKBGhvc3QSKGNvcnRleC1ndy5sb2tpLWRldi1zc2Quc3ZjLmNsdXN0ZXIubG9jYWwKPQoIaW5zdGFuY2USMXByb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ejpwcm9tdGFpbDpodHRwLW1ldHJpY3MKKQoDam9iEiJsb2tpLWRldi1zc2QvcHJvbXRhaWwtbG9raS1kZXYtc3NkChkKCW5hbWVzcGFjZRIMbG9raS1kZXYtc3NkCiIKA3BvZBIbcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6ChgKBnJlYXNvbhIOaW5nZXN0ZXJfZXJyb3ISBxDig/ickDIK+gIKMAoIX19uYW1lX18SJGNvcnRleF9yZXNwb25zZV9tZXNzYWdlX2J5dGVzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChEKAmxlEgsyLjYyMTQ0ZSswOAoNCgZtZXRob2QSA0dFVAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKFgoFcm91dGUSDWluZ2VzdGVyX3JpbmcSEAkAAAAAAADwPxDygvickDIKvwMKOgoIX19uYW1lX18SLnRoYW5vc19jYWNoZV9vcGVyYXRpb25fZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKFAoHYmFja2VuZBIJbWVtY2FjaGVkChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29tcG9uZW50EgtkaXN0cmlidXRvcgoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoLCgJsZRIFMC4wMDEKHwoEbmFtZRIXYWdncmVnYXRpb24tcnVsZXMtY2FjaGUKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKFQoJb3BlcmF0aW9uEghnZXRtdWx0aQojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAAAAAIG7QBDygvickDIKtQMKOgoIX19uYW1lX18SLnRoYW5vc19jYWNoZV9vcGVyYXRpb25fZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKFAoHYmFja2VuZBIJbWVtY2FjaGVkChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29tcG9uZW50EgtkaXN0cmlidXRvcgoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoHCgJsZRIBMQofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoYCglvcGVyYXRpb24SC2NvbXBhcmVzd2FwCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIHEPKC+JyQMgrNAgo3CghfX25hbWVfXxIrY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX2V4ZW1wbGFyc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxMDc5MRIHEPKC+JyQMgrGAgowCghfX25hbWVfXxIkY29ydGV4X2Rpc3RyaWJ1dG9yX21ldGFkYXRhX2luX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTEzNzU3EgcQ8oL4nJAyCrUDCjUKCF9fbmFtZV9fEiljb3J0ZXhfa3ZfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCiAKB2t2X25hbWUSFWRpc3RyaWJ1dG9yLWhhdHJhY2tlcgoHCgJsZRIBMQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCglvcGVyYXRpb24SA0NBUwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDwoEcm9sZRIHcHJpbWFyeQoSCgtzdGF0dXNfY29kZRIDMjAwCgwKBHR5cGUSBGV0Y2QSEAkAAAAAAFKnQBDygvickDIKzQIKNwoIX19uYW1lX18SK2NvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9leGVtcGxhcnNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTA3OTUSBxDygvickDIKgAQKMQoIX19uYW1lX18SJWtlcGxlcl9jb250YWluZXJfcGFja2FnZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDE0NTY3NzUxYjU2ZDUzOTU3ZjZiMjdiNmEwYTU2Zjg4MjkxZjA5ZWMwNjE3MWE3NzhlZGRhZGQyNjE4OTIwYWIKHwoOY29udGFpbmVyX25hbWUSDWlwLW1hc3EtYWdlbnQKIgoTY29udGFpbmVyX25hbWVzcGFjZRILa3ViZS1zeXN0ZW0KNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh8KCHBvZF9uYW1lEhNpcC1tYXNxLWFnZW50LTc5N2NuCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCZhuEoPAMmFAEPiD+JyQMgq6AgoqCghfX25hbWVfXxIeZ29fbWVtc3RhdHNfbWNhY2hlX2ludXNlX2J5dGVzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAADAwkAQ8oL4nJAyCr0DCjkKCF9fbmFtZV9fEi10aGFub3NfY2FjaGVfb3BlcmF0aW9uX2RhdGFfc2l6ZV9ieXRlc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKFAoHYmFja2VuZBIJbWVtY2FjaGVkChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29tcG9uZW50EgtkaXN0cmlidXRvcgoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoKCgJsZRIEMzIuMAofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoVCglvcGVyYXRpb24SCGdldG11bHRpCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAAAAO8NAEPKC+JyQMgrrAgo3CghfX25hbWVfXxIra2Fma2Ffd3JpdGVyX3Byb2R1Y2VfY29tcHJlc3NlZF9ieXRlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoiCgV0b3BpYxIZbWltaXItZGV2LTEzLWFnZ3JlZ2F0aW9ucxIQCQAAWA8S3R5CEPKC+JyQMgrTAgoiCghfX25hbWVfXxIWZ29fZ2NfZHVyYXRpb25fc2Vjb25kcwofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoQCghxdWFudGlsZRIEMC4yNRIQCQoq4igEaBc/ENqC+JyQMgr6AgowCghfX25hbWVfXxIkY29ydGV4X3Jlc3BvbnNlX21lc3NhZ2VfYnl0ZXNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKEQoCbGUSCzUuMjQyODhlKzA2Cg0KBm1ldGhvZBIDR0VUChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoWCgVyb3V0ZRINaW5nZXN0ZXJfcmluZxIQCQAAAAAAAPA/EPKC+JyQMgrdAgo3CghfX25hbWVfXxIrY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX2V4ZW1wbGFyc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAodCgR1c2VyEhVtaW1pci1jb250aW51b3VzLXRlc3QSBxDygvickDIK7AIKKAoIX19uYW1lX18SHHByb210YWlsX2VuY29kZWRfYnl0ZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoVCgljb250YWluZXISCHByb210YWlsCjAKBGhvc3QSKGNvcnRleC1ndy5sb2tpLWRldi1zc2Quc3ZjLmNsdXN0ZXIubG9jYWwKPQoIaW5zdGFuY2USMXByb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ejpwcm9tdGFpbDpodHRwLW1ldHJpY3MKKQoDam9iEiJsb2tpLWRldi1zc2QvcHJvbXRhaWwtbG9raS1kZXYtc3NkChkKCW5hbWVzcGFjZRIMbG9raS1kZXYtc3NkCiIKA3BvZBIbcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6EhAJAAAAAADIkkAQ4oP4nJAyCoUECi4KCF9fbmFtZV9fEiJrZXBsZXJfY29udGFpbmVyX2NvcmVfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkBhZGEzYmJjZjJjZmU3ZWQ4ODMwODYyZDRiOTkwN2Y1MTc4YTgzMmEwYTZjNTUyMTdhZDZkYzMzMTg4MTM4ZGY1Ch0KDmNvbnRhaW5lcl9uYW1lEgtkaXN0cmlidXRvcgojChNjb250YWluZXJfbmFtZXNwYWNlEgxtaW1pci1kZXYtMTEKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCigKCHBvZF9uYW1lEhxkaXN0cmlidXRvci02OTY5N2ZjY2NiLXF6emZiCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCbpJDAIrxyRAEPiD+JyQMgrUAgo1CghfX25hbWVfXxIpY29ydGV4X2Rpc3RyaWJ1dG9yX3JlY2VpdmVkX3NhbXBsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTM3NTcSEAkAAABwxQN7QRDygvickDIK7wMKMAoIX19uYW1lX18SJGtlcGxlcl9jb250YWluZXJfYnBmX2Jsb2NrX2lycV90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAOWFkODM0MzliODhiOTg1YzgwY2IxNzUyMzk2ZDIzZDljZTQxMTQyY2YwOTcyZTA0OGE5MDk1OTMzOGIzZWI1MwokCg5jb250YWluZXJfbmFtZRISbmV0ZmlsdGVyLWV4cG9ydGVyCikKE2NvbnRhaW5lcl9uYW1lc3BhY2USEm5ldGZpbHRlci1leHBvcnRlcgo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCiQKCHBvZF9uYW1lEhhuZXRmaWx0ZXItZXhwb3J0ZXItZnh0aDcKDQoGc291cmNlEgNicGYSEAkAAAAAAAAQQBD4g/ickDIKzwIKMAoIX19uYW1lX18SJGNvcnRleF9kaXN0cmlidXRvcl9yZXF1ZXN0c19pbl90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxNDEzNRIQCQAAAAAAABBAEPKC+JyQMgq1Awo6CghfX25hbWVfXxIudGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgcKAmxlEgEzCh8KBG5hbWUSF2FnZ3JlZ2F0aW9uLXJ1bGVzLWNhY2hlChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChgKCW9wZXJhdGlvbhILY29tcGFyZXN3YXAKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEgcQ8oL4nJAyCvoDCi4KCF9fbmFtZV9fEiJrZXBsZXJfY29udGFpbmVyX2RyYW1fam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkBlNTZjMzcwYjJiYzkwMmJjMDRmMmNkY2Q0Njg4ZjFlZDU5ZDFiNDE1ZDEwZGE5ZjQzM2NkNzcxMDk0NGJkZDZhCh8KDmNvbnRhaW5lcl9uYW1lEg1nY2UtcGQtZHJpdmVyCiIKE2NvbnRhaW5lcl9uYW1lc3BhY2USC2t1YmUtc3lzdGVtCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgocCghwb2RfbmFtZRIQcGRjc2ktbm9kZS1iaHN4agodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSEAlkO99PjZf0PxD4g/ickDIKtQMKOgoIX19uYW1lX18SLnRoYW5vc19jYWNoZV9vcGVyYXRpb25fZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKFAoHYmFja2VuZBIJbWVtY2FjaGVkChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29tcG9uZW50EgtkaXN0cmlidXRvcgoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoHCgJsZRIBNgofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoYCglvcGVyYXRpb24SC2NvbXBhcmVzd2FwCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIHEPKC+JyQMgqXAwo0CghfX25hbWVfXxIocHJvbXRhaWxfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChUKCWNvbnRhaW5lchIIcHJvbXRhaWwKMAoEaG9zdBIoY29ydGV4LWd3Lmxva2ktZGV2LXNzZC5zdmMuY2x1c3Rlci5sb2NhbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKCQoCbGUSAzAuNQoZCgluYW1lc3BhY2USDGxva2ktZGV2LXNzZAoiCgNwb2QSG3Byb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6egoSCgtzdGF0dXNfY29kZRIDMjA0EhAJAAAAAAAA8D8Q4oP4nJAyCuICCkMKCF9fbmFtZV9fEjdjb3J0ZXhfZGlzdHJpYnV0b3JfbGF0ZXN0X3NlZW5fc2FtcGxlX3RpbWVzdGFtcF9zZWNvbmRzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTE3ODU1EhAJNV5iIzeq2UEQ8oL4nJAyCs4CCi8KCF9fbmFtZV9fEiNjb3J0ZXhfZGlzdHJpYnV0b3Jfc2FtcGxlc19pbl90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxNjU0MxIQCQAAANA33n5BEPKC+JyQMgriAgpDCghfX25hbWVfXxI3Y29ydGV4X2Rpc3RyaWJ1dG9yX2xhdGVzdF9zZWVuX3NhbXBsZV90aW1lc3RhbXBfc2Vjb25kcwofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxOTEwMRIQCSuHdgE3qtlBEPKC+JyQMgqkAwpCCghfX25hbWVfXxI2Y29ydGV4X2luZ2VzdGVyX2NsaWVudF9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCwoCbGUSBTQuMDk2ChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiIKCW9wZXJhdGlvbhIVL2NvcnRleC5Jbmdlc3Rlci9QdXNoCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAofCgtzdGF0dXNfY29kZRIQRGVhZGxpbmVFeGNlZWRlZBIQCQAAAADARtBAEPKC+JyQMgq/Awo6CghfX25hbWVfXxIudGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUwLjAwNQofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoVCglvcGVyYXRpb24SCGdldG11bHRpCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAAAAFsNAEPKC+JyQMgqmAwoyCghfX25hbWVfXxImY29ydGV4X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoKCgJsZRIEMC4yNQoOCgZtZXRob2QSBGdSUEMKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCiYKBXJvdXRlEh0vZGlzdHJpYnV0b3IuRGlzdHJpYnV0b3IvUHVzaAoRCgtzdGF0dXNfY29kZRICT0sKCwoCd3MSBWZhbHNlEhAJAAAAAAAH5EAQ8oL4nJAyCs0CCioKCF9fbmFtZV9fEh5rYWZrYV93cml0ZXJfd3JpdGVfYnl0ZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoRCgdub2RlX2lkEgZzZWVkXzAKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAACAR0AQ8oL4nJAyCuECCjIKCF9fbmFtZV9fEiZrYWZrYV93cml0ZXJfd3JpdGVfdGltZV9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUwLjAwNAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoQCgdub2RlX2lkEgUyMDAwMAojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAAAAmDUVQRDygvickDIK1QIKNgoIX19uYW1lX18SKmNvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9yZXF1ZXN0c190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxMDkxMxIQCQAAAAAATNRAEPKC+JyQMgrIAwouCghfX25hbWVfXxIia2VwbGVyX2NvbnRhaW5lcl9jb3JlX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyCiAKDGNvbnRhaW5lcl9pZBIQa2VybmVsX3Byb2Nlc3NlcwoiCg5jb250YWluZXJfbmFtZRIQa2VybmVsX3Byb2Nlc3NlcwodChNjb250YWluZXJfbmFtZXNwYWNlEgZrZXJuZWwKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyChwKCHBvZF9uYW1lEhBrZXJuZWxfcHJvY2Vzc2VzCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCfp+arx001RAEPiD+JyQMgrtAgopCghfX25hbWVfXxIdcmVzcG9uc2VfbWVzc2FnZV9ieXRlc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoVCgljb250YWluZXISCHByb210YWlsCj0KCGluc3RhbmNlEjFwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42eno6cHJvbXRhaWw6aHR0cC1tZXRyaWNzCikKA2pvYhIibG9raS1kZXYtc3NkL3Byb210YWlsLWxva2ktZGV2LXNzZAoRCgJsZRILMi42MjE0NGUrMDYKDQoGbWV0aG9kEgNHRVQKGQoJbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKIgoDcG9kEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42enoKDgoFcm91dGUSBXJlYWR5EhAJAAAAAADsrkAQ4oP4nJAyCs8DCjAKCF9fbmFtZV9fEiRrZXBsZXJfY29udGFpbmVyX2JwZl9ibG9ja19pcnFfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQGJkYjU1ZDkyYjZmZjM4NTBkYTljYmFhODJlNGJlMTU2MDY4YzMzNzNlMWY2OGUzYWU5NGRkMGU4YjRhOGIyNjkKFwoOY29udGFpbmVyX25hbWUSBWFnZW50CiMKE2NvbnRhaW5lcl9uYW1lc3BhY2USDHByb210YWlsLW9wcwo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyChcKCHBvZF9uYW1lEgthZ2VudC1ncHMyZgoNCgZzb3VyY2USA2JwZhIQCQAAAAAAgEJAEPiD+JyQMgqvAwo3CghfX25hbWVfXxIrdGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kdXJhdGlvbl9zZWNvbmRzX3N1bQofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCh8KBG5hbWUSF2FnZ3JlZ2F0aW9uLXJ1bGVzLWNhY2hlChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzChUKCW9wZXJhdGlvbhIIZ2V0bXVsdGkKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJ03JBX3VxJUAQ8oL4nJAyCtQCCi8KCF9fbmFtZV9fEiNqYWVnZXJfdHJhY2VyX3NhbXBsZXJfdXBkYXRlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZgoNCgZyZXN1bHQSA2VychIHENqC+JyQMgq3Awo6CghfX25hbWVfXxIudGhhbm9zX2NhY2hlX29wZXJhdGlvbl9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAoUCgdiYWNrZW5kEgltZW1jYWNoZWQKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb21wb25lbnQSC2Rpc3RyaWJ1dG9yChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCgsKAmxlEgUwLjAwNQofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoWCglvcGVyYXRpb24SCWluY3JlbWVudAojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASBxDygvickDIKwAMKPgoIX19uYW1lX18SMnRoYW5vc19jYWNoZV9nZXRtdWx0aV9nYXRlX2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChQKB2JhY2tlbmQSCW1lbWNhY2hlZAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbXBvbmVudBILZGlzdHJpYnV0b3IKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKCQoCbGUSAzcyMAofCgRuYW1lEhdhZ2dyZWdhdGlvbi1ydWxlcy1jYWNoZQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoUCgdvdXRjb21lEglwZXJtaXR0ZWQKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAAA7w0AQ8oL4nJAyCrkDCjUKCF9fbmFtZV9fEiljb3J0ZXhfa3ZfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCiAKB2t2X25hbWUSFWRpc3RyaWJ1dG9yLWhhdHJhY2tlcgoKCgJsZRIEMC4wNQoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwoRCglvcGVyYXRpb24SBExpc3QKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg8KBHJvbGUSB3ByaW1hcnkKEgoLc3RhdHVzX2NvZGUSAzIwMAoMCgR0eXBlEgRldGNkEhAJAAAAAAAgc0AQ8oL4nJAyCvsDCi8KCF9fbmFtZV9fEiNrZXBsZXJfY29udGFpbmVyX290aGVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAZTU2YzM3MGIyYmM5MDJiYzA0ZjJjZGNkNDY4OGYxZWQ1OWQxYjQxNWQxMGRhOWY0MzNjZDc3MTA5NDRiZGQ2YQofCg5jb250YWluZXJfbmFtZRINZ2NlLXBkLWRyaXZlcgoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHAoIcG9kX25hbWUSEHBkY3NpLW5vZGUtYmhzeGoKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJrkfhehSu7z8Q+IP4nJAyCokDCjoKCF9fbmFtZV9fEi5ncmFmYW5hY29tX2FwaV9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKCwoEY29kZRIDMjAwChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyCgoKAmxlEgQxMC4wChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzChUKBHBhdGgSDWhvc3RlZC10cmFjZXMKJgoDcG9kEh9sb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmEhAJAAAAAABIt0AQ2oL4nJAyCsMCCiQKCF9fbmFtZV9fEhhnb19tZW1zdGF0c19oZWFwX29iamVjdHMKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAobCgljb250YWluZXISDmxvZ3MtZm9yd2FyZGVyCkcKCGluc3RhbmNlEjtsb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmOmxvZ3MtZm9yd2FyZGVyOmh0dHAtbWV0cmljcwoiCgNqb2ISG2luc2lnaHQtbG9ncy9sb2dzLWZvcndhcmRlcgoZCgluYW1lc3BhY2USDGluc2lnaHQtbG9ncwomCgNwb2QSH2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2YSEAkAAAAAQHbpQBDagvickDIKtQIKJQoIX19uYW1lX18SGWdvX21lbXN0YXRzX25leHRfZ2NfYnl0ZXMKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAABQ2+bmQRDygvickDIKzwIKSAoIX19uYW1lX18SPGNvcnRleF9oYV90cmFja2VyX3JlcGxpY2FzX2NsZWFudXBfbWFya2VkX2Zvcl9kZWxldGlvbl90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIHEPKC+JyQMgrVAwo1CghfX25hbWVfXxIpa2VwbGVyX2NvbnRhaW5lcl9icGZfcGFnZV9jYWNoZV9oaXRfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDdmZTlhMDcxNjZmZDc4YWI5NjcxNjljMzI0OTRmODMzNTE2NWRlMWRlOTQ1ZWIxZGFmM2Y4YzgyNWRhN2RjZGMKGgoOY29udGFpbmVyX25hbWUSCGluZ2VzdGVyCiMKE2NvbnRhaW5lcl9uYW1lc3BhY2USDG1pbWlyLWRldi0xNAo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh4KCHBvZF9uYW1lEhJpbmdlc3Rlci16b25lLWItMTEKDQoGc291cmNlEgNicGYSBxD4g/ickDIKxwIKKwoIX19uYW1lX18SH2NvcnRleF9sYWJlbHNfcGVyX3NhbXBsZV9idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoKCgJsZRIEMTUuMAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAANC7PM8NQhDygvickDIKjAQKMQoIX19uYW1lX18SJWtlcGxlcl9jb250YWluZXJfcGFja2FnZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDFhNWRhYjY0NTc5YjNlYjI4MjI5OTYwZTUwNjRjNTYzYmM5Mjc1MmQwMTdjYWVhOGI3OWMxNDNhMmNjNTczMTYKHwoOY29udGFpbmVyX25hbWUSDXJ1bGVyLXF1ZXJpZXIKIwoTY29udGFpbmVyX25hbWVzcGFjZRIMbWltaXItZGV2LTExCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgoqCghwb2RfbmFtZRIecnVsZXItcXVlcmllci01OTk0ZDRiNzVjLWZiYjhuCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCQAAAAAsWu9AEPiD+JyQMgrwAgo/CghfX25hbWVfXxIzY29ydGV4X2Rpc3RyaWJ1dG9yX2luc3RhbmNlX3JlamVjdGVkX3JlcXVlc3RzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCigKBnJlYXNvbhIeZGlzdHJpYnV0b3JfbWF4X2luZ2VzdGlvbl9yYXRlEgcQ8oL4nJAyCqgECjEKCF9fbmFtZV9fEiVrZXBsZXJfY29udGFpbmVyX3BhY2thZ2Vfam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkA3YzEwYTBlZmEzNDE0NDcwNjVmMmQ5NTIwYWVjODU5MGUyMzAwYzkyMjk3NWRiZDA3NDljNDE3YzBmZTg2OGJiChwKDmNvbnRhaW5lcl9uYW1lEgprdWJlLXByb3h5CiIKE2NvbnRhaW5lcl9uYW1lc3BhY2USC2t1YmUtc3lzdGVtCjUKCGluc3RhbmNlEilrZXBsZXItdmRqdzI6a2VwbGVyLWV4cG9ydGVyOmh0dHAtbWV0cmljcwoUCgNqb2ISDWtlcGxlci9rZXBsZXIKDwoEbW9kZRIHZHluYW1pYwoTCgluYW1lc3BhY2USBmtlcGxlcgoTCgNwb2QSDGtlcGxlci12ZGp3MgpKCghwb2RfbmFtZRI+a3ViZS1wcm94eS1na2UtZGV2LXVzLWNlbnRyYWwtMC1tYWluLW4yczE2LTMtMWRkLTliNTAyZDk2LXR2bWsKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJwcqhRTZ9nUAQ+IP4nJAyCvsDCjIKCF9fbmFtZV9fEiZrZXBsZXJfY29udGFpbmVyX3BsYXRmb3JtX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAZTU2YzM3MGIyYmM5MDJiYzA0ZjJjZGNkNDY4OGYxZWQ1OWQxYjQxNWQxMGRhOWY0MzNjZDc3MTA5NDRiZGQ2YQofCg5jb250YWluZXJfbmFtZRINZ2NlLXBkLWRyaXZlcgoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHAoIcG9kX25hbWUSEHBkY3NpLW5vZGUtYmhzeGoKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJexSuR+EulkAQ+IP4nJAyCsYCCjAKCF9fbmFtZV9fEiRjb3J0ZXhfZGlzdHJpYnV0b3JfbWV0YWRhdGFfaW5fdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTM2NzcSBxDygvickDIKzAIKNgoIX19uYW1lX18SKmNvcnRleF9kaXN0cmlidXRvcl9yZWNlaXZlZF9tZXRhZGF0YV90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoNCgR1c2VyEgUxOTEwMRIHEPKC+JyQMgrSAwopCghfX25hbWVfXxIda2VwbGVyX2NvbnRhaW5lcl9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDE4ZjhiMjRmYTdhYjA5NmVlYTc2MDk5MDUzYzBhNWYzNDQzMTAxMTYzNzY3YjVlZmQyYWFkZjFhMzg1NGRlNGUKHAoOY29udGFpbmVyX25hbWUSCmdvbGRwaW5nZXIKIQoTY29udGFpbmVyX25hbWVzcGFjZRIKZ29sZHBpbmdlcgo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKHAoIcG9kX25hbWUSEGdvbGRwaW5nZXItNjhsOXMSEAnJdr6fGi/9PxD4g/ickDIK2AMKLQoIX19uYW1lX18SIWtlcGxlcl9jb250YWluZXJfZ3B1X2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAOGI3YzQwM2E2ZGFkMmFlOTA1NjU5MzIzZDQyZTNlYTQ4MTJiNmI3NTk1NTFkMWFiMjA5ZmEyMGZmN2RhNDc1ZQohCg5jb250YWluZXJfbmFtZRIPa2VwbGVyLWV4cG9ydGVyCh0KE2NvbnRhaW5lcl9uYW1lc3BhY2USBmtlcGxlcgo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKGAoIcG9kX25hbWUSDGtlcGxlci12ZGp3MgoPCgZzb3VyY2USBWR1bW15EgcQ+IP4nJAyCuICCkMKCF9fbmFtZV9fEjdjb3J0ZXhfZGlzdHJpYnV0b3JfbGF0ZXN0X3NlZW5fc2FtcGxlX3RpbWVzdGFtcF9zZWNvbmRzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTE5MzIxEhAJ7nzfKjeq2UEQ8oL4nJAyCvMDCi4KCF9fbmFtZV9fEiJrZXBsZXJfY29udGFpbmVyX2RyYW1fam91bGVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKHAoJY29udGFpbmVyEg9rZXBsZXItZXhwb3J0ZXIKUAoMY29udGFpbmVyX2lkEkAxOGY4YjI0ZmE3YWIwOTZlZWE3NjA5OTA1M2MwYTVmMzQ0MzEwMTE2Mzc2N2I1ZWZkMmFhZGYxYTM4NTRkZTRlChwKDmNvbnRhaW5lcl9uYW1lEgpnb2xkcGluZ2VyCiEKE2NvbnRhaW5lcl9uYW1lc3BhY2USCmdvbGRwaW5nZXIKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyChwKCHBvZF9uYW1lEhBnb2xkcGluZ2VyLTY4bDlzCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCXNoke18XzRAEPiD+JyQMgqWAwoyCghfX25hbWVfXxImY29ydGV4X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoKCgJsZRIEMC4yNQoNCgZtZXRob2QSA0dFVAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKFgoFcm91dGUSDWluZ2VzdGVyX3JpbmcKEgoLc3RhdHVzX2NvZGUSAzIwMAoLCgJ3cxIFZmFsc2USEAkAAAAAAADwPxDygvickDIKiwMKNAoIX19uYW1lX18SKGNvcnRleF9kaXN0cmlidXRvcl9leGVtcGxhcnNfcGVyX3JlcXVlc3QKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDAoEdXNlchIEOTk2MCJJCMzBkQMZAAAAoA10e0EpnZeMzwi6WyUwn+vNAloCEAxiIKaUE4Ag6L0G9oIGjZgKx84NlfcDiZQCwbUBgQmuGawpePKC+JyQMgrSAgosCghfX25hbWVfXxIgY29ydGV4X3JlcXVlc3RfbWVzc2FnZV9ieXRlc19zdW0KHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoNCgZtZXRob2QSA0dFVAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDgoFcm91dGUSBXJlYWR5EgcQ8oL4nJAyCqYECi8KCF9fbmFtZV9fEiNrZXBsZXJfY29udGFpbmVyX290aGVyX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAN2MxMGEwZWZhMzQxNDQ3MDY1ZjJkOTUyMGFlYzg1OTBlMjMwMGM5MjI5NzVkYmQwNzQ5YzQxN2MwZmU4NjhiYgocCg5jb250YWluZXJfbmFtZRIKa3ViZS1wcm94eQoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCg8KBG1vZGUSB2R5bmFtaWMKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKSgoIcG9kX25hbWUSPmt1YmUtcHJveHktZ2tlLWRldi11cy1jZW50cmFsLTAtbWFpbi1uMnMxNi0zLTFkZC05YjUwMmQ5Ni10dm1rCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIQCbByaJHtoHBAEPiD+JyQMgqJAwo6CghfX25hbWVfXxIuZ3JhZmFuYWNvbV9hcGlfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wCgsKBGNvZGUSAzIwMAobCgljb250YWluZXISDmxvZ3MtZm9yd2FyZGVyCkcKCGluc3RhbmNlEjtsb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmOmxvZ3MtZm9yd2FyZGVyOmh0dHAtbWV0cmljcwoiCgNqb2ISG2luc2lnaHQtbG9ncy9sb2dzLWZvcndhcmRlcgoKCgJsZRIEMTAuMAoZCgluYW1lc3BhY2USDGluc2lnaHQtbG9ncwoVCgRwYXRoEg1ob3N0ZWQtYWxlcnRzCiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZhIQCQAAAAAAqLdAENqC+JyQMgr3AwoyCghfX25hbWVfXxIma2VwbGVyX2NvbnRhaW5lcl9wbGF0Zm9ybV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDhiN2M0MDNhNmRhZDJhZTkwNTY1OTMyM2Q0MmUzZWE0ODEyYjZiNzU5NTUxZDFhYjIwOWZhMjBmZjdkYTQ3NWUKIQoOY29udGFpbmVyX25hbWUSD2tlcGxlci1leHBvcnRlcgodChNjb250YWluZXJfbmFtZXNwYWNlEgZrZXBsZXIKNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoPCgRtb2RlEgdkeW5hbWljChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyChgKCHBvZF9uYW1lEgxrZXBsZXItdmRqdzIKHQoGc291cmNlEhN0cmFpbmVkX3Bvd2VyX21vZGVsEhAJMzMzMzMzsz8Q+IP4nJAyCtoDCjUKCF9fbmFtZV9fEilrZXBsZXJfY29udGFpbmVyX2JwZl9wYWdlX2NhY2hlX2hpdF90b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAMTQ1Njc3NTFiNTZkNTM5NTdmNmIyN2I2YTBhNTZmODgyOTFmMDllYzA2MTcxYTc3OGVkZGFkZDI2MTg5MjBhYgofCg5jb250YWluZXJfbmFtZRINaXAtbWFzcS1hZ2VudAoiChNjb250YWluZXJfbmFtZXNwYWNlEgtrdWJlLXN5c3RlbQo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCh8KCHBvZF9uYW1lEhNpcC1tYXNxLWFnZW50LTc5N2NuCg0KBnNvdXJjZRIDYnBmEgcQ+IP4nJAyCs0CCjcKCF9fbmFtZV9fEitjb3J0ZXhfZGlzdHJpYnV0b3JfcmVjZWl2ZWRfZXhlbXBsYXJzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTEwNzg5EgcQ8oL4nJAyCtcCCjgKCF9fbmFtZV9fEixjb3J0ZXhfc2VsZl9tb25pdG9yaW5nX2Rpc2tfcmVhZF9ieXRlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKDQoGZGV2aWNlEgNzZGoKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcBIQCQAAAACQI5hBEPKC+JyQMgq5AgopCghfX25hbWVfXxIdZ29fbWVtc3RhdHNfYWxsb2NfYnl0ZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAQM8j47q0QhDygvickDIK4gMKMAoIX19uYW1lX18SJGtlcGxlcl9jb250YWluZXJfdW5jb3JlX2pvdWxlc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPa2VwbGVyLWV4cG9ydGVyClAKDGNvbnRhaW5lcl9pZBJAZjlkMTE0NzY3NzhiNmYwYzhhOWYwOTBjZjVhODk2MjViODQwNjA2N2M3MTZkZWJiZWUwYzQ2ZGY0NmYzNGNiNwoXCg5jb250YWluZXJfbmFtZRIFYWdlbnQKIQoTY29udGFpbmVyX25hbWVzcGFjZRIKYWdlbnQtbG9ncwo1CghpbnN0YW5jZRIpa2VwbGVyLXZkancyOmtlcGxlci1leHBvcnRlcjpodHRwLW1ldHJpY3MKFAoDam9iEg1rZXBsZXIva2VwbGVyCgwKBG1vZGUSBGlkbGUKEwoJbmFtZXNwYWNlEgZrZXBsZXIKEwoDcG9kEgxrZXBsZXItdmRqdzIKFwoIcG9kX25hbWUSC2FnZW50LWpxOTlrCh0KBnNvdXJjZRITdHJhaW5lZF9wb3dlcl9tb2RlbBIHEPiD+JyQMgrjAgooCghfX25hbWVfXxIccHJvbXRhaWxfYmF0Y2hfcmV0cmllc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChUKCWNvbnRhaW5lchIIcHJvbXRhaWwKMAoEaG9zdBIoY29ydGV4LWd3Lmxva2ktZGV2LXNzZC5zdmMuY2x1c3Rlci5sb2NhbAo9CghpbnN0YW5jZRIxcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6OnByb210YWlsOmh0dHAtbWV0cmljcwopCgNqb2ISImxva2ktZGV2LXNzZC9wcm9tdGFpbC1sb2tpLWRldi1zc2QKGQoJbmFtZXNwYWNlEgxsb2tpLWRldi1zc2QKIgoDcG9kEhtwcm9tdGFpbC1sb2tpLWRldi1zc2QtbG42enoSBxDig/ickDIKtwMKNQoIX19uYW1lX18SKWNvcnRleF9rdl9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKIAoHa3ZfbmFtZRIVZGlzdHJpYnV0b3ItaGF0cmFja2VyCgkKAmxlEgMyLjUKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKEAoJb3BlcmF0aW9uEgNDQVMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg8KBHJvbGUSB3ByaW1hcnkKEgoLc3RhdHVzX2NvZGUSAzIwMAoMCgR0eXBlEgRldGNkEhAJAAAAAABSp0AQ8oL4nJAyCp0DCkIKCF9fbmFtZV9fEjZjb3J0ZXhfaW5nZXN0ZXJfY2xpZW50X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19idWNrZXQKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoLCgJsZRIFMC4yNTYKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKKQoJb3BlcmF0aW9uEhwvZ3JwYy5oZWFsdGgudjEuSGVhbHRoL0NoZWNrCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoRCgtzdGF0dXNfY29kZRICT0sSEAkAAACAhkRGQRDygvickDIKjQMKOAoIX19uYW1lX18SLGNvcnRleF9zZWxmX21vbml0b3JpbmdfZmlsZXN5c3RlbV9maWxlc19mcmVlCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgoTCgZkZXZpY2USCS9kZXYvc2RhMQoOCgZmc3R5cGUSBGV4dDQKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yCh4KCm1vdW50cG9pbnQSEC9ldGMvcmVzb2x2LmNvbmYKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAGz7SUEQ8oL4nJAyCt0CCg4KCF9fbmFtZV9fEgJ1cAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChwKCWNvbnRhaW5lchIPc3RhdGljLWV4cG9ydGVyClcKCGluc3RhbmNlEktweXJvc2NvcGUtYnVja2V0LWluZm8tZXhwb3J0ZXItN2ZkYjQ0NDY0LWJ4ZmJ0OnN0YXRpYy1leHBvcnRlcjpodHRwLW1ldHJpY3MKMgoDam9iEitmaXJlLWRldi0wMDEvcHlyb3Njb3BlLWJ1Y2tldC1pbmZvLWV4cG9ydGVyChkKCW5hbWVzcGFjZRIMZmlyZS1kZXYtMDAxCjUKA3BvZBIucHlyb3Njb3BlLWJ1Y2tldC1pbmZvLWV4cG9ydGVyLTdmZGI0NDQ2NC1ieGZidBIQCQAAAAAAAPA/ENuD+JyQMgrkAgoyCghfX25hbWVfXxImbmV0X2Nvbm50cmFja19kaWFsZXJfY29ubl9mYWlsZWRfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yChYKC2RpYWxlcl9uYW1lEgd1bmtub3duCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKEQoGcmVhc29uEgd0aW1lb3V0EgcQ8oL4nJAyCtkCCjoKCF9fbmFtZV9fEi5jb3J0ZXhfc2VsZl9tb25pdG9yaW5nX2Rpc2tfcmVhZHNfbWVyZ2VkX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgoNCgZkZXZpY2USA3NkYgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwEhAJAAAAAACSrUAQ8oL4nJAyCvoCCjAKCF9fbmFtZV9fEiRjb3J0ZXhfaGFfdHJhY2tlcl9rdl9zdG9yZV9jYXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCiEKEGV4cG9ydGVkX2NsdXN0ZXISDWRldi1ldS13ZXN0LTIKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoVCgR1c2VyEg1iaWxsaW5nLXN1cGVyEhAJAAAAAACQg0AQ8oL4nJAyCtgCCjoKCF9fbmFtZV9fEi5jb3J0ZXhfZGlzdHJpYnV0b3Jfc2FtcGxlX2RlbGF5X3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKDAoCbGUSBjM2MDAuMAoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3ASEAkAAGiL6nIOQhDygvickDIK0QIKOwoIX19uYW1lX18SL2xva2lfbG9ncWxfcXVlcnlzdGF0c19pbmdlc3Rlcl9zZW50X2xpbmVzX3RvdGFsCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGwoJY29udGFpbmVyEg5sb2dzLWZvcndhcmRlcgpHCghpbnN0YW5jZRI7bG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZjpsb2dzLWZvcndhcmRlcjpodHRwLW1ldHJpY3MKIgoDam9iEhtpbnNpZ2h0LWxvZ3MvbG9ncy1mb3J3YXJkZXIKGQoJbmFtZXNwYWNlEgxpbnNpZ2h0LWxvZ3MKJgoDcG9kEh9sb2dzLWZvcndhcmRlci03NzZkNTY3YzliLThocWNmEgcQ2oL4nJAyCuICCkMKCF9fbmFtZV9fEjdjb3J0ZXhfZGlzdHJpYnV0b3JfbGF0ZXN0X3NlZW5fc2FtcGxlX3RpbWVzdGFtcF9zZWNvbmRzCh8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKGAoJY29udGFpbmVyEgtkaXN0cmlidXRvcgpBCghpbnN0YW5jZRI1ZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcDpkaXN0cmlidXRvcjpodHRwLW1ldHJpY3MKHwoDam9iEhhtaW1pci1kZXYtMTMvZGlzdHJpYnV0b3IKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg0KBHVzZXISBTEzNzU3EhAJAAAABTeq2UEQ8oL4nJAyCpsECjAKCF9fbmFtZV9fEiRrZXBsZXJfY29udGFpbmVyX3VuY29yZV9qb3VsZXNfdG90YWwKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAocCgljb250YWluZXISD2tlcGxlci1leHBvcnRlcgpQCgxjb250YWluZXJfaWQSQDdjMTBhMGVmYTM0MTQ0NzA2NWYyZDk1MjBhZWM4NTkwZTIzMDBjOTIyOTc1ZGJkMDc0OWM0MTdjMGZlODY4YmIKHAoOY29udGFpbmVyX25hbWUSCmt1YmUtcHJveHkKIgoTY29udGFpbmVyX25hbWVzcGFjZRILa3ViZS1zeXN0ZW0KNQoIaW5zdGFuY2USKWtlcGxlci12ZGp3MjprZXBsZXItZXhwb3J0ZXI6aHR0cC1tZXRyaWNzChQKA2pvYhINa2VwbGVyL2tlcGxlcgoMCgRtb2RlEgRpZGxlChMKCW5hbWVzcGFjZRIGa2VwbGVyChMKA3BvZBIMa2VwbGVyLXZkancyCkoKCHBvZF9uYW1lEj5rdWJlLXByb3h5LWdrZS1kZXYtdXMtY2VudHJhbC0wLW1haW4tbjJzMTYtMy0xZGQtOWI1MDJkOTYtdHZtawodCgZzb3VyY2USE3RyYWluZWRfcG93ZXJfbW9kZWwSBxD4g/ickDIK7gIKNAoIX19uYW1lX18SKGNvcnRleF9kaXN0cmlidXRvcl9leGVtcGxhcnNfcGVyX3JlcXVlc3QKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTA4MDkiKwhhGQAAAACAzMRAKZ2XjM8IulslMElaBAgKEAZiBgQDAgQICHjygvickDIKxQIKPAoIX19uYW1lX18SMGphZWdlcl90cmFjZXJfc3Bhbl9jb250ZXh0X2RlY29kaW5nX2Vycm9yc190b3RhbAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChUKCWNvbnRhaW5lchIIcHJvbXRhaWwKPQoIaW5zdGFuY2USMXByb210YWlsLWxva2ktZGV2LXNzZC1sbjZ6ejpwcm9tdGFpbDpodHRwLW1ldHJpY3MKKQoDam9iEiJsb2tpLWRldi1zc2QvcHJvbXRhaWwtbG9raS1kZXYtc3NkChkKCW5hbWVzcGFjZRIMbG9raS1kZXYtc3NkCiIKA3BvZBIbcHJvbXRhaWwtbG9raS1kZXYtc3NkLWxuNnp6EgcQ4oP4nJAyCrkDCjUKCF9fbmFtZV9fEiljb3J0ZXhfa3ZfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChsKB2t2X25hbWUSEGRpc3RyaWJ1dG9yLXJpbmcKCgoCbGUSBDAuMjUKGQoJbmFtZXNwYWNlEgxtaW1pci1kZXYtMTMKEAoJb3BlcmF0aW9uEgNHRVQKIwoDcG9kEhxkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwCg8KBHJvbGUSB3ByaW1hcnkKEgoLc3RhdHVzX2NvZGUSAzIwMAoSCgR0eXBlEgptZW1iZXJsaXN0EhAJAAAAAAAA8D8Q8oL4nJAyCooDCjoKCF9fbmFtZV9fEi5ncmFmYW5hY29tX2FwaV9yZXF1ZXN0X2R1cmF0aW9uX3NlY29uZHNfYnVja2V0Ch8KC2Fzc2VydHNfZW52EhBkZXYtdXMtY2VudHJhbC0wChsKB2NsdXN0ZXISEGRldi11cy1jZW50cmFsLTAKCwoEY29kZRIDMjAwChsKCWNvbnRhaW5lchIObG9ncy1mb3J3YXJkZXIKRwoIaW5zdGFuY2USO2xvZ3MtZm9yd2FyZGVyLTc3NmQ1NjdjOWItOGhxY2Y6bG9ncy1mb3J3YXJkZXI6aHR0cC1tZXRyaWNzCiIKA2pvYhIbaW5zaWdodC1sb2dzL2xvZ3MtZm9yd2FyZGVyCgoKAmxlEgQrSW5mChkKCW5hbWVzcGFjZRIMaW5zaWdodC1sb2dzChYKBHBhdGgSDjx1bnJlY29nbmlzZWQ+CiYKA3BvZBIfbG9ncy1mb3J3YXJkZXItNzc2ZDU2N2M5Yi04aHFjZhIQCQAAAAAASLdAENqC+JyQMgrzAgoyCghfX25hbWVfXxImY29ydGV4X2Rpc3RyaWJ1dG9yX3NhbXBsZXNfcGVyX3JlcXVlc3QKHwoLYXNzZXJ0c19lbnYSEGRldi11cy1jZW50cmFsLTAKGwoHY2x1c3RlchIQZGV2LXVzLWNlbnRyYWwtMAoYCgljb250YWluZXISC2Rpc3RyaWJ1dG9yCkEKCGluc3RhbmNlEjVkaXN0cmlidXRvci03Nzc0ZDRmNmZiLXNzNmtwOmRpc3RyaWJ1dG9yOmh0dHAtbWV0cmljcwofCgNqb2ISGG1pbWlyLWRldi0xMy9kaXN0cmlidXRvcgoZCgluYW1lc3BhY2USDG1pbWlyLWRldi0xMwojCgNwb2QSHGRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3AKDQoEdXNlchIFMTkxMDEiMgjsARkAAAAA4ETqQCmdl4zPCLpbJTAAWgIQDGIOAhJCVTIxAEKCAcMBCiB48oL4nJAyCpEDCkEKCF9fbmFtZV9fEjVjb3J0ZXhfaW5nZXN0ZXJfY2xpZW50X3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19jb3VudAofCgthc3NlcnRzX2VudhIQZGV2LXVzLWNlbnRyYWwtMAobCgdjbHVzdGVyEhBkZXYtdXMtY2VudHJhbC0wChgKCWNvbnRhaW5lchILZGlzdHJpYnV0b3IKQQoIaW5zdGFuY2USNWRpc3RyaWJ1dG9yLTc3NzRkNGY2ZmItc3M2a3A6ZGlzdHJpYnV0b3I6aHR0cC1tZXRyaWNzCh8KA2pvYhIYbWltaXItZGV2LTEzL2Rpc3RyaWJ1dG9yChkKCW5hbWVzcGFjZRIMbWltaXItZGV2LTEzCiIKCW9wZXJhdGlvbhIVL2NvcnRleC5Jbmdlc3Rlci9QdXNoCiMKA3BvZBIcZGlzdHJpYnV0b3ItNzc3NGQ0ZjZmYi1zczZrcAoaCgtzdGF0dXNfY29kZRILVW5hdmFpbGFibGUSEAkAAAAAAJDbQBDygvickDIaJQgCEiFqYWVnZXJfdHJhY2VyX3N0YXJ0ZWRfc3BhbnNfdG90YWwaJggCEiJqYWVnZXJfdHJhY2VyX3JlcG9ydGVyX3NwYW5zX3RvdGFsGiIIAhIebG9nc19mb3J3YXJkZXJfdGNwX2Nvbm5lY3Rpb25zGiAIAhIcZ29fbWVtc3RhdHNfaGVhcF9hbGxvY19ieXRlcxoxCAISLWxvZ3NfZm9yd2FyZGVyX3JlcXVlc3RfZHVyYXRpb25fc2Vjb25kc19jb3VudBowCAISLGxvZ3NfZm9yd2FyZGVyX2h0dHBfY29ubmVjdGlvbnNfY2xvc2VkX3RvdGFsGiEIAhIdZ29fbWVtc3RhdHNfYWxsb2NfYnl0ZXNfdG90YWwaGggCEhZnb19nY19kdXJhdGlvbl9zZWNvbmRzGjIIAhIubG9nc19mb3J3YXJkZXJfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldBodCAISGWdvX21lbXN0YXRzX25leHRfZ2NfYnl0ZXMaJQgCEiFqYWVnZXJfdHJhY2VyX3N0YXJ0ZWRfc3BhbnNfdG90YWwaOQgCEjVtZW1iZXJsaXN0X3RjcF90cmFuc3BvcnRfb3V0Z29pbmdfc3RyZWFtX2Vycm9yc190b3RhbBo9CAISOWNvcnRleF9kaXN0cmlidXRvcl9mb3J3YXJkaW5nX3J1bGVzX3JlbG9hZF9mYWlsdXJlc190b3RhbBpACAISPGNvcnRleF9oYV90cmFja2VyX3JlcGxpY2FzX2NsZWFudXBfbWFya2VkX2Zvcl9kZWxldGlvbl90b3RhbBovCAISK2thZmthX3dyaXRlcl9idWZmZXJlZF9wcm9kdWNlX3JlY29yZHNfdG90YWwaFQgCEhFjb3J0ZXhfYnVpbGRfaW5mbxolCAISIXRpbWVyX21lbWJlcmxpc3RfcHVzaFB1bGxOb2RlX3N1bRo6CAISNmNvcnRleF9pbmdlc3Rlcl9jbGllbnRfcmVxdWVzdF9kdXJhdGlvbl9zZWNvbmRzX2J1Y2tldBoaCAISFmdvX2djX2R1cmF0aW9uX3NlY29uZHMaIAgCEhxnb19tZW1zdGF0c19oZWFwX2FsbG9jX2J5dGVzGiIIAhIeY29ydGV4X2Rpc2NhcmRlZF9zYW1wbGVzX3RvdGFsGjQIAhIwY29ydGV4X2Rpc3RyaWJ1dG9yX25vbl9oYV9zYW1wbGVzX3JlY2VpdmVkX3RvdGFsGjsIAhI3bWltaXJfZGlzdHJpYnV0b3JfYWdncmVnYXRpb25zX2ZvcndhcmRpbmdfc2FtcGxlc190b3RhbBotCAISKWNvcnRleF9kaXN0cmlidXRvcl9pbmdlc3Rfc3RvcmFnZV9lbmFibGVkGioIAhIma2Fma2Ffd3JpdGVyX3dyaXRlX3RpbWVfc2Vjb25kc19idWNrZXQaKQgCEiVncmFmYW5hX21ldHJpY3NfZW50ZXJwcmlzZV9idWlsZF9pbmZvGiYIAhIiamFlZ2VyX3RyYWNlcl9yZXBvcnRlcl9zcGFuc190b3RhbBofCAISG2thZmthX3dyaXRlcl9jb25uZWN0c190b3RhbBodCAISGWdvX21lbXN0YXRzX25leHRfZ2NfYnl0ZXMaIQgCEh1nb19tZW1zdGF0c19hbGxvY19ieXRlc190b3RhbBovCAISK2xva2lfY2FuYXJ5X3Jlc3BvbnNlX2xhdGVuY3lfc2Vjb25kc19idWNrZXQaKAgCEiRsb2tpX2NhbmFyeV9zcG90X2NoZWNrX2VudHJpZXNfdG90YWwaIAgCEhxnb19tZW1zdGF0c19oZWFwX2FsbG9jX2J5dGVzGhoIAhIWZ29fZ2NfZHVyYXRpb25fc2Vjb25kcxohCAISHWdvX21lbXN0YXRzX2FsbG9jX2J5dGVzX3RvdGFsGh0IAhIZZ29fbWVtc3RhdHNfbmV4dF9nY19ieXRlcxolCAISIWphZWdlcl90cmFjZXJfc3RhcnRlZF9zcGFuc190b3RhbBoYCAISFGxva2lfbG9nX2ZsdXNoZXNfc3VtGhkIAhIVdGNwX2Nvbm5lY3Rpb25zX2xpbWl0GiAIAhIccHJvbXRhaWxfbXV0YXRlZF9ieXRlc190b3RhbBoaCAISFmdvX2djX2R1cmF0aW9uX3NlY29uZHMaIQgCEh1nb19tZW1zdGF0c19hbGxvY19ieXRlc190b3RhbBogCAISHGdvX21lbXN0YXRzX2hlYXBfYWxsb2NfYnl0ZXMaJggCEiJqYWVnZXJfdHJhY2VyX3JlcG9ydGVyX3NwYW5zX3RvdGFsGh0IAhIZZ29fbWVtc3RhdHNfbmV4dF9nY19ieXRlcw==",
  "Headers": [
    {
      "Key": "traceparent",
      "Value": "MDAtMDAwMDAwMDAwMDAwMDAwMDYwMmE1YTNmODM5MDc2N2UtNTBlOTNhYjU4Y2NlYjAxZi0wMA=="
    }
  ],
  "Timestamp": "2024-07-30T12:29:32.697Z",
  "Topic": "ingest",
  "Partition": 0,
  "Attrs": {},
  "ProducerEpoch": 0,
  "ProducerID": 0,
  "LeaderEpoch": 0,
  "Offset": 434850004,
  "Context": null
}

```

</p>
</details>